// @hrbolek/uoisfrontend-shared/dist/es/index.js
import * as U from "react";
import Ee, { useContext as Ke, useMemo as tn, useState as fe, useCallback as Ce, useRef as he, useEffect as Se, createContext as Oo, useLayoutEffect as ko, cloneElement as Nt, forwardRef as Rr, useImperativeHandle as Do } from "react";
import rn from "react-dom";
import { createSlice as wt, configureStore as Ao } from "@reduxjs/toolkit";
function Ct(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var dr = { exports: {} }, Nn = {};
/**
 * @license React
 * react-jsx-dev-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Br;
function Vo() {
  return Br || (Br = 1, process.env.NODE_ENV !== "production" && function() {
    var e = Ee, n = Symbol.for("react.element"), r = Symbol.for("react.portal"), t = Symbol.for("react.fragment"), o = Symbol.for("react.strict_mode"), i = Symbol.for("react.profiler"), u = Symbol.for("react.provider"), s = Symbol.for("react.context"), c = Symbol.for("react.forward_ref"), f = Symbol.for("react.suspense"), m = Symbol.for("react.suspense_list"), v = Symbol.for("react.memo"), h = Symbol.for("react.lazy"), C = Symbol.for("react.offscreen"), S = Symbol.iterator, T = "@@iterator";
    function O(l) {
      if (l === null || typeof l != "object")
        return null;
      var g = S && l[S] || l[T];
      return typeof g == "function" ? g : null;
    }
    var A = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function _(l) {
      {
        for (var g = arguments.length, R = new Array(g > 1 ? g - 1 : 0), a = 1; a < g; a++)
          R[a - 1] = arguments[a];
        B("error", l, R);
      }
    }
    function B(l, g, R) {
      {
        var a = A.ReactDebugCurrentFrame, p = a.getStackAddendum();
        p !== "" && (g += "%s", R = R.concat([p]));
        var b = R.map(function(x) {
          return String(x);
        });
        b.unshift("Warning: " + g), Function.prototype.apply.call(console[l], console, b);
      }
    }
    var q = !1, E = !1, G = !1, se = !1, ye = !1, xe;
    xe = Symbol.for("react.module.reference");
    function we(l) {
      return !!(typeof l == "string" || typeof l == "function" || l === t || l === i || ye || l === o || l === f || l === m || se || l === C || q || E || G || typeof l == "object" && l !== null && (l.$$typeof === h || l.$$typeof === v || l.$$typeof === u || l.$$typeof === s || l.$$typeof === c || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      l.$$typeof === xe || l.getModuleId !== void 0));
    }
    function be(l, g, R) {
      var a = l.displayName;
      if (a)
        return a;
      var p = g.displayName || g.name || "";
      return p !== "" ? R + "(" + p + ")" : R;
    }
    function ge(l) {
      return l.displayName || "Context";
    }
    function Z(l) {
      if (l == null)
        return null;
      if (typeof l.tag == "number" && _("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof l == "function")
        return l.displayName || l.name || null;
      if (typeof l == "string")
        return l;
      switch (l) {
        case t:
          return "Fragment";
        case r:
          return "Portal";
        case i:
          return "Profiler";
        case o:
          return "StrictMode";
        case f:
          return "Suspense";
        case m:
          return "SuspenseList";
      }
      if (typeof l == "object")
        switch (l.$$typeof) {
          case s:
            var g = l;
            return ge(g) + ".Consumer";
          case u:
            var R = l;
            return ge(R._context) + ".Provider";
          case c:
            return be(l, l.render, "ForwardRef");
          case v:
            var a = l.displayName || null;
            return a !== null ? a : Z(l.type) || "Memo";
          case h: {
            var p = l, b = p._payload, x = p._init;
            try {
              return Z(x(b));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var te = Object.assign, le = 0, ne, pe, oe, Re, y, w, M;
    function V() {
    }
    V.__reactDisabledLog = !0;
    function $() {
      {
        if (le === 0) {
          ne = console.log, pe = console.info, oe = console.warn, Re = console.error, y = console.group, w = console.groupCollapsed, M = console.groupEnd;
          var l = {
            configurable: !0,
            enumerable: !0,
            value: V,
            writable: !0
          };
          Object.defineProperties(console, {
            info: l,
            log: l,
            warn: l,
            error: l,
            group: l,
            groupCollapsed: l,
            groupEnd: l
          });
        }
        le++;
      }
    }
    function J() {
      {
        if (le--, le === 0) {
          var l = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: te({}, l, {
              value: ne
            }),
            info: te({}, l, {
              value: pe
            }),
            warn: te({}, l, {
              value: oe
            }),
            error: te({}, l, {
              value: Re
            }),
            group: te({}, l, {
              value: y
            }),
            groupCollapsed: te({}, l, {
              value: w
            }),
            groupEnd: te({}, l, {
              value: M
            })
          });
        }
        le < 0 && _("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var I = A.ReactCurrentDispatcher, W;
    function Y(l, g, R) {
      {
        if (W === void 0)
          try {
            throw Error();
          } catch (p) {
            var a = p.stack.trim().match(/\n( *(at )?)/);
            W = a && a[1] || "";
          }
        return `
` + W + l;
      }
    }
    var H = !1, L;
    {
      var ue = typeof WeakMap == "function" ? WeakMap : Map;
      L = new ue();
    }
    function N(l, g) {
      if (!l || H)
        return "";
      {
        var R = L.get(l);
        if (R !== void 0)
          return R;
      }
      var a;
      H = !0;
      var p = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var b;
      b = I.current, I.current = null, $();
      try {
        if (g) {
          var x = function() {
            throw Error();
          };
          if (Object.defineProperty(x.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(x, []);
            } catch (me) {
              a = me;
            }
            Reflect.construct(l, [], x);
          } else {
            try {
              x.call();
            } catch (me) {
              a = me;
            }
            l.call(x.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (me) {
            a = me;
          }
          l();
        }
      } catch (me) {
        if (me && a && typeof me.stack == "string") {
          for (var j = me.stack.split(`
`), z = a.stack.split(`
`), D = j.length - 1, k = z.length - 1; D >= 1 && k >= 0 && j[D] !== z[k]; )
            k--;
          for (; D >= 1 && k >= 0; D--, k--)
            if (j[D] !== z[k]) {
              if (D !== 1 || k !== 1)
                do
                  if (D--, k--, k < 0 || j[D] !== z[k]) {
                    var ie = `
` + j[D].replace(" at new ", " at ");
                    return l.displayName && ie.includes("<anonymous>") && (ie = ie.replace("<anonymous>", l.displayName)), typeof l == "function" && L.set(l, ie), ie;
                  }
                while (D >= 1 && k >= 0);
              break;
            }
        }
      } finally {
        H = !1, I.current = b, J(), Error.prepareStackTrace = p;
      }
      var ae = l ? l.displayName || l.name : "", ce = ae ? Y(ae) : "";
      return typeof l == "function" && L.set(l, ce), ce;
    }
    function ve(l, g, R) {
      return N(l, !1);
    }
    function Ne(l) {
      var g = l.prototype;
      return !!(g && g.isReactComponent);
    }
    function K(l, g, R) {
      if (l == null)
        return "";
      if (typeof l == "function")
        return N(l, Ne(l));
      if (typeof l == "string")
        return Y(l);
      switch (l) {
        case f:
          return Y("Suspense");
        case m:
          return Y("SuspenseList");
      }
      if (typeof l == "object")
        switch (l.$$typeof) {
          case c:
            return ve(l.render);
          case v:
            return K(l.type, g, R);
          case h: {
            var a = l, p = a._payload, b = a._init;
            try {
              return K(b(p), g, R);
            } catch {
            }
          }
        }
      return "";
    }
    var je = Object.prototype.hasOwnProperty, ke = {}, We = A.ReactDebugCurrentFrame;
    function De(l) {
      if (l) {
        var g = l._owner, R = K(l.type, l._source, g ? g.type : null);
        We.setExtraStackFrame(R);
      } else
        We.setExtraStackFrame(null);
    }
    function sn(l, g, R, a, p) {
      {
        var b = Function.call.bind(je);
        for (var x in l)
          if (b(l, x)) {
            var j = void 0;
            try {
              if (typeof l[x] != "function") {
                var z = Error((a || "React class") + ": " + R + " type `" + x + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof l[x] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw z.name = "Invariant Violation", z;
              }
              j = l[x](g, x, a, R, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (D) {
              j = D;
            }
            j && !(j instanceof Error) && (De(p), _("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", a || "React class", R, x, typeof j), De(null)), j instanceof Error && !(j.message in ke) && (ke[j.message] = !0, De(p), _("Failed %s type: %s", R, j.message), De(null));
          }
      }
    }
    var un = Array.isArray;
    function Pe(l) {
      return un(l);
    }
    function Xe(l) {
      {
        var g = typeof Symbol == "function" && Symbol.toStringTag, R = g && l[Symbol.toStringTag] || l.constructor.name || "Object";
        return R;
      }
    }
    function ln(l) {
      try {
        return X(l), !1;
      } catch {
        return !0;
      }
    }
    function X(l) {
      return "" + l;
    }
    function Oe(l) {
      if (ln(l))
        return _("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", Xe(l)), X(l);
    }
    var _e = A.ReactCurrentOwner, In = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, cn, fn, Ue;
    Ue = {};
    function $n(l) {
      if (je.call(l, "ref")) {
        var g = Object.getOwnPropertyDescriptor(l, "ref").get;
        if (g && g.isReactWarning)
          return !1;
      }
      return l.ref !== void 0;
    }
    function Ln(l) {
      if (je.call(l, "key")) {
        var g = Object.getOwnPropertyDescriptor(l, "key").get;
        if (g && g.isReactWarning)
          return !1;
      }
      return l.key !== void 0;
    }
    function Bn(l, g) {
      if (typeof l.ref == "string" && _e.current && g && _e.current.stateNode !== g) {
        var R = Z(_e.current.type);
        Ue[R] || (_('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', Z(_e.current.type), l.ref), Ue[R] = !0);
      }
    }
    function Wn(l, g) {
      {
        var R = function() {
          cn || (cn = !0, _("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", g));
        };
        R.isReactWarning = !0, Object.defineProperty(l, "key", {
          get: R,
          configurable: !0
        });
      }
    }
    function Un(l, g) {
      {
        var R = function() {
          fn || (fn = !0, _("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", g));
        };
        R.isReactWarning = !0, Object.defineProperty(l, "ref", {
          get: R,
          configurable: !0
        });
      }
    }
    var Yn = function(l, g, R, a, p, b, x) {
      var j = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: n,
        // Built-in properties that belong on the element
        type: l,
        key: g,
        ref: R,
        props: x,
        // Record the component responsible for creating this element.
        _owner: b
      };
      return j._store = {}, Object.defineProperty(j._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(j, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: a
      }), Object.defineProperty(j, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: p
      }), Object.freeze && (Object.freeze(j.props), Object.freeze(j)), j;
    };
    function qn(l, g, R, a, p) {
      {
        var b, x = {}, j = null, z = null;
        R !== void 0 && (Oe(R), j = "" + R), Ln(g) && (Oe(g.key), j = "" + g.key), $n(g) && (z = g.ref, Bn(g, p));
        for (b in g)
          je.call(g, b) && !In.hasOwnProperty(b) && (x[b] = g[b]);
        if (l && l.defaultProps) {
          var D = l.defaultProps;
          for (b in D)
            x[b] === void 0 && (x[b] = D[b]);
        }
        if (j || z) {
          var k = typeof l == "function" ? l.displayName || l.name || "Unknown" : l;
          j && Wn(x, k), z && Un(x, k);
        }
        return Yn(l, j, z, p, a, _e.current, x);
      }
    }
    var Ye = A.ReactCurrentOwner, dn = A.ReactDebugCurrentFrame;
    function Te(l) {
      if (l) {
        var g = l._owner, R = K(l.type, l._source, g ? g.type : null);
        dn.setExtraStackFrame(R);
      } else
        dn.setExtraStackFrame(null);
    }
    var qe;
    qe = !1;
    function He(l) {
      return typeof l == "object" && l !== null && l.$$typeof === n;
    }
    function mn() {
      {
        if (Ye.current) {
          var l = Z(Ye.current.type);
          if (l)
            return `

Check the render method of \`` + l + "`.";
        }
        return "";
      }
    }
    function Hn(l) {
      {
        if (l !== void 0) {
          var g = l.fileName.replace(/^.*[\\\/]/, ""), R = l.lineNumber;
          return `

Check your code at ` + g + ":" + R + ".";
        }
        return "";
      }
    }
    var pn = {};
    function zn(l) {
      {
        var g = mn();
        if (!g) {
          var R = typeof l == "string" ? l : l.displayName || l.name;
          R && (g = `

Check the top-level render call using <` + R + ">.");
        }
        return g;
      }
    }
    function vn(l, g) {
      {
        if (!l._store || l._store.validated || l.key != null)
          return;
        l._store.validated = !0;
        var R = zn(g);
        if (pn[R])
          return;
        pn[R] = !0;
        var a = "";
        l && l._owner && l._owner !== Ye.current && (a = " It was passed a child from " + Z(l._owner.type) + "."), Te(l), _('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', R, a), Te(null);
      }
    }
    function hn(l, g) {
      {
        if (typeof l != "object")
          return;
        if (Pe(l))
          for (var R = 0; R < l.length; R++) {
            var a = l[R];
            He(a) && vn(a, g);
          }
        else if (He(l))
          l._store && (l._store.validated = !0);
        else if (l) {
          var p = O(l);
          if (typeof p == "function" && p !== l.entries)
            for (var b = p.call(l), x; !(x = b.next()).done; )
              He(x.value) && vn(x.value, g);
        }
      }
    }
    function Jn(l) {
      {
        var g = l.type;
        if (g == null || typeof g == "string")
          return;
        var R;
        if (typeof g == "function")
          R = g.propTypes;
        else if (typeof g == "object" && (g.$$typeof === c || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        g.$$typeof === v))
          R = g.propTypes;
        else
          return;
        if (R) {
          var a = Z(g);
          sn(R, l.props, "prop", a, l);
        } else if (g.PropTypes !== void 0 && !qe) {
          qe = !0;
          var p = Z(g);
          _("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", p || "Unknown");
        }
        typeof g.getDefaultProps == "function" && !g.getDefaultProps.isReactClassApproved && _("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function Gn(l) {
      {
        for (var g = Object.keys(l.props), R = 0; R < g.length; R++) {
          var a = g[R];
          if (a !== "children" && a !== "key") {
            Te(l), _("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", a), Te(null);
            break;
          }
        }
        l.ref !== null && (Te(l), _("Invalid attribute `ref` supplied to `React.Fragment`."), Te(null));
      }
    }
    function xn(l, g, R, a, p, b) {
      {
        var x = we(l);
        if (!x) {
          var j = "";
          (l === void 0 || typeof l == "object" && l !== null && Object.keys(l).length === 0) && (j += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var z = Hn(p);
          z ? j += z : j += mn();
          var D;
          l === null ? D = "null" : Pe(l) ? D = "array" : l !== void 0 && l.$$typeof === n ? (D = "<" + (Z(l.type) || "Unknown") + " />", j = " Did you accidentally export a JSX literal instead of a component?") : D = typeof l, _("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", D, j);
        }
        var k = qn(l, g, R, p, b);
        if (k == null)
          return k;
        if (x) {
          var ie = g.children;
          if (ie !== void 0)
            if (a)
              if (Pe(ie)) {
                for (var ae = 0; ae < ie.length; ae++)
                  hn(ie[ae], l);
                Object.freeze && Object.freeze(ie);
              } else
                _("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              hn(ie, l);
        }
        return l === t ? Gn(k) : Jn(k), k;
      }
    }
    var Kn = xn;
    Nn.Fragment = t, Nn.jsxDEV = Kn;
  }()), Nn;
}
var wn = {};
/**
 * @license React
 * react-jsx-dev-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Wr;
function Po() {
  if (Wr)
    return wn;
  Wr = 1;
  var e = Symbol.for("react.fragment");
  return wn.Fragment = e, wn.jsxDEV = void 0, wn;
}
process.env.NODE_ENV === "production" ? dr.exports = Po() : dr.exports = Vo();
var d = dr.exports, Rt = { exports: {} };
/*!
	Copyright (c) 2018 Jed Watson.
	Licensed under the MIT License (MIT), see
	http://jedwatson.github.io/classnames
*/
(function(e) {
  (function() {
    var n = {}.hasOwnProperty;
    function r() {
      for (var t = [], o = 0; o < arguments.length; o++) {
        var i = arguments[o];
        if (i) {
          var u = typeof i;
          if (u === "string" || u === "number")
            t.push(i);
          else if (Array.isArray(i)) {
            if (i.length) {
              var s = r.apply(null, i);
              s && t.push(s);
            }
          } else if (u === "object") {
            if (i.toString !== Object.prototype.toString && !i.toString.toString().includes("[native code]")) {
              t.push(i.toString());
              continue;
            }
            for (var c in i)
              n.call(i, c) && i[c] && t.push(c);
          }
        }
      }
      return t.join(" ");
    }
    e.exports ? (r.default = r, e.exports = r) : window.classNames = r;
  })();
})(Rt);
var Mo = Rt.exports;
const re = /* @__PURE__ */ Ct(Mo);
var mr = { exports: {} }, bn = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ur;
function Io() {
  return Ur || (Ur = 1, process.env.NODE_ENV !== "production" && function() {
    var e = Ee, n = Symbol.for("react.element"), r = Symbol.for("react.portal"), t = Symbol.for("react.fragment"), o = Symbol.for("react.strict_mode"), i = Symbol.for("react.profiler"), u = Symbol.for("react.provider"), s = Symbol.for("react.context"), c = Symbol.for("react.forward_ref"), f = Symbol.for("react.suspense"), m = Symbol.for("react.suspense_list"), v = Symbol.for("react.memo"), h = Symbol.for("react.lazy"), C = Symbol.for("react.offscreen"), S = Symbol.iterator, T = "@@iterator";
    function O(a) {
      if (a === null || typeof a != "object")
        return null;
      var p = S && a[S] || a[T];
      return typeof p == "function" ? p : null;
    }
    var A = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function _(a) {
      {
        for (var p = arguments.length, b = new Array(p > 1 ? p - 1 : 0), x = 1; x < p; x++)
          b[x - 1] = arguments[x];
        B("error", a, b);
      }
    }
    function B(a, p, b) {
      {
        var x = A.ReactDebugCurrentFrame, j = x.getStackAddendum();
        j !== "" && (p += "%s", b = b.concat([j]));
        var z = b.map(function(D) {
          return String(D);
        });
        z.unshift("Warning: " + p), Function.prototype.apply.call(console[a], console, z);
      }
    }
    var q = !1, E = !1, G = !1, se = !1, ye = !1, xe;
    xe = Symbol.for("react.module.reference");
    function we(a) {
      return !!(typeof a == "string" || typeof a == "function" || a === t || a === i || ye || a === o || a === f || a === m || se || a === C || q || E || G || typeof a == "object" && a !== null && (a.$$typeof === h || a.$$typeof === v || a.$$typeof === u || a.$$typeof === s || a.$$typeof === c || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      a.$$typeof === xe || a.getModuleId !== void 0));
    }
    function be(a, p, b) {
      var x = a.displayName;
      if (x)
        return x;
      var j = p.displayName || p.name || "";
      return j !== "" ? b + "(" + j + ")" : b;
    }
    function ge(a) {
      return a.displayName || "Context";
    }
    function Z(a) {
      if (a == null)
        return null;
      if (typeof a.tag == "number" && _("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof a == "function")
        return a.displayName || a.name || null;
      if (typeof a == "string")
        return a;
      switch (a) {
        case t:
          return "Fragment";
        case r:
          return "Portal";
        case i:
          return "Profiler";
        case o:
          return "StrictMode";
        case f:
          return "Suspense";
        case m:
          return "SuspenseList";
      }
      if (typeof a == "object")
        switch (a.$$typeof) {
          case s:
            var p = a;
            return ge(p) + ".Consumer";
          case u:
            var b = a;
            return ge(b._context) + ".Provider";
          case c:
            return be(a, a.render, "ForwardRef");
          case v:
            var x = a.displayName || null;
            return x !== null ? x : Z(a.type) || "Memo";
          case h: {
            var j = a, z = j._payload, D = j._init;
            try {
              return Z(D(z));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var te = Object.assign, le = 0, ne, pe, oe, Re, y, w, M;
    function V() {
    }
    V.__reactDisabledLog = !0;
    function $() {
      {
        if (le === 0) {
          ne = console.log, pe = console.info, oe = console.warn, Re = console.error, y = console.group, w = console.groupCollapsed, M = console.groupEnd;
          var a = {
            configurable: !0,
            enumerable: !0,
            value: V,
            writable: !0
          };
          Object.defineProperties(console, {
            info: a,
            log: a,
            warn: a,
            error: a,
            group: a,
            groupCollapsed: a,
            groupEnd: a
          });
        }
        le++;
      }
    }
    function J() {
      {
        if (le--, le === 0) {
          var a = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: te({}, a, {
              value: ne
            }),
            info: te({}, a, {
              value: pe
            }),
            warn: te({}, a, {
              value: oe
            }),
            error: te({}, a, {
              value: Re
            }),
            group: te({}, a, {
              value: y
            }),
            groupCollapsed: te({}, a, {
              value: w
            }),
            groupEnd: te({}, a, {
              value: M
            })
          });
        }
        le < 0 && _("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var I = A.ReactCurrentDispatcher, W;
    function Y(a, p, b) {
      {
        if (W === void 0)
          try {
            throw Error();
          } catch (j) {
            var x = j.stack.trim().match(/\n( *(at )?)/);
            W = x && x[1] || "";
          }
        return `
` + W + a;
      }
    }
    var H = !1, L;
    {
      var ue = typeof WeakMap == "function" ? WeakMap : Map;
      L = new ue();
    }
    function N(a, p) {
      if (!a || H)
        return "";
      {
        var b = L.get(a);
        if (b !== void 0)
          return b;
      }
      var x;
      H = !0;
      var j = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var z;
      z = I.current, I.current = null, $();
      try {
        if (p) {
          var D = function() {
            throw Error();
          };
          if (Object.defineProperty(D.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(D, []);
            } catch (Me) {
              x = Me;
            }
            Reflect.construct(a, [], D);
          } else {
            try {
              D.call();
            } catch (Me) {
              x = Me;
            }
            a.call(D.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (Me) {
            x = Me;
          }
          a();
        }
      } catch (Me) {
        if (Me && x && typeof Me.stack == "string") {
          for (var k = Me.stack.split(`
`), ie = x.stack.split(`
`), ae = k.length - 1, ce = ie.length - 1; ae >= 1 && ce >= 0 && k[ae] !== ie[ce]; )
            ce--;
          for (; ae >= 1 && ce >= 0; ae--, ce--)
            if (k[ae] !== ie[ce]) {
              if (ae !== 1 || ce !== 1)
                do
                  if (ae--, ce--, ce < 0 || k[ae] !== ie[ce]) {
                    var me = `
` + k[ae].replace(" at new ", " at ");
                    return a.displayName && me.includes("<anonymous>") && (me = me.replace("<anonymous>", a.displayName)), typeof a == "function" && L.set(a, me), me;
                  }
                while (ae >= 1 && ce >= 0);
              break;
            }
        }
      } finally {
        H = !1, I.current = z, J(), Error.prepareStackTrace = j;
      }
      var Ze = a ? a.displayName || a.name : "", Lr = Ze ? Y(Ze) : "";
      return typeof a == "function" && L.set(a, Lr), Lr;
    }
    function ve(a, p, b) {
      return N(a, !1);
    }
    function Ne(a) {
      var p = a.prototype;
      return !!(p && p.isReactComponent);
    }
    function K(a, p, b) {
      if (a == null)
        return "";
      if (typeof a == "function")
        return N(a, Ne(a));
      if (typeof a == "string")
        return Y(a);
      switch (a) {
        case f:
          return Y("Suspense");
        case m:
          return Y("SuspenseList");
      }
      if (typeof a == "object")
        switch (a.$$typeof) {
          case c:
            return ve(a.render);
          case v:
            return K(a.type, p, b);
          case h: {
            var x = a, j = x._payload, z = x._init;
            try {
              return K(z(j), p, b);
            } catch {
            }
          }
        }
      return "";
    }
    var je = Object.prototype.hasOwnProperty, ke = {}, We = A.ReactDebugCurrentFrame;
    function De(a) {
      if (a) {
        var p = a._owner, b = K(a.type, a._source, p ? p.type : null);
        We.setExtraStackFrame(b);
      } else
        We.setExtraStackFrame(null);
    }
    function sn(a, p, b, x, j) {
      {
        var z = Function.call.bind(je);
        for (var D in a)
          if (z(a, D)) {
            var k = void 0;
            try {
              if (typeof a[D] != "function") {
                var ie = Error((x || "React class") + ": " + b + " type `" + D + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof a[D] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw ie.name = "Invariant Violation", ie;
              }
              k = a[D](p, D, x, b, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (ae) {
              k = ae;
            }
            k && !(k instanceof Error) && (De(j), _("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", x || "React class", b, D, typeof k), De(null)), k instanceof Error && !(k.message in ke) && (ke[k.message] = !0, De(j), _("Failed %s type: %s", b, k.message), De(null));
          }
      }
    }
    var un = Array.isArray;
    function Pe(a) {
      return un(a);
    }
    function Xe(a) {
      {
        var p = typeof Symbol == "function" && Symbol.toStringTag, b = p && a[Symbol.toStringTag] || a.constructor.name || "Object";
        return b;
      }
    }
    function ln(a) {
      try {
        return X(a), !1;
      } catch {
        return !0;
      }
    }
    function X(a) {
      return "" + a;
    }
    function Oe(a) {
      if (ln(a))
        return _("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", Xe(a)), X(a);
    }
    var _e = A.ReactCurrentOwner, In = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, cn, fn, Ue;
    Ue = {};
    function $n(a) {
      if (je.call(a, "ref")) {
        var p = Object.getOwnPropertyDescriptor(a, "ref").get;
        if (p && p.isReactWarning)
          return !1;
      }
      return a.ref !== void 0;
    }
    function Ln(a) {
      if (je.call(a, "key")) {
        var p = Object.getOwnPropertyDescriptor(a, "key").get;
        if (p && p.isReactWarning)
          return !1;
      }
      return a.key !== void 0;
    }
    function Bn(a, p) {
      if (typeof a.ref == "string" && _e.current && p && _e.current.stateNode !== p) {
        var b = Z(_e.current.type);
        Ue[b] || (_('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', Z(_e.current.type), a.ref), Ue[b] = !0);
      }
    }
    function Wn(a, p) {
      {
        var b = function() {
          cn || (cn = !0, _("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", p));
        };
        b.isReactWarning = !0, Object.defineProperty(a, "key", {
          get: b,
          configurable: !0
        });
      }
    }
    function Un(a, p) {
      {
        var b = function() {
          fn || (fn = !0, _("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", p));
        };
        b.isReactWarning = !0, Object.defineProperty(a, "ref", {
          get: b,
          configurable: !0
        });
      }
    }
    var Yn = function(a, p, b, x, j, z, D) {
      var k = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: n,
        // Built-in properties that belong on the element
        type: a,
        key: p,
        ref: b,
        props: D,
        // Record the component responsible for creating this element.
        _owner: z
      };
      return k._store = {}, Object.defineProperty(k._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(k, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: x
      }), Object.defineProperty(k, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: j
      }), Object.freeze && (Object.freeze(k.props), Object.freeze(k)), k;
    };
    function qn(a, p, b, x, j) {
      {
        var z, D = {}, k = null, ie = null;
        b !== void 0 && (Oe(b), k = "" + b), Ln(p) && (Oe(p.key), k = "" + p.key), $n(p) && (ie = p.ref, Bn(p, j));
        for (z in p)
          je.call(p, z) && !In.hasOwnProperty(z) && (D[z] = p[z]);
        if (a && a.defaultProps) {
          var ae = a.defaultProps;
          for (z in ae)
            D[z] === void 0 && (D[z] = ae[z]);
        }
        if (k || ie) {
          var ce = typeof a == "function" ? a.displayName || a.name || "Unknown" : a;
          k && Wn(D, ce), ie && Un(D, ce);
        }
        return Yn(a, k, ie, j, x, _e.current, D);
      }
    }
    var Ye = A.ReactCurrentOwner, dn = A.ReactDebugCurrentFrame;
    function Te(a) {
      if (a) {
        var p = a._owner, b = K(a.type, a._source, p ? p.type : null);
        dn.setExtraStackFrame(b);
      } else
        dn.setExtraStackFrame(null);
    }
    var qe;
    qe = !1;
    function He(a) {
      return typeof a == "object" && a !== null && a.$$typeof === n;
    }
    function mn() {
      {
        if (Ye.current) {
          var a = Z(Ye.current.type);
          if (a)
            return `

Check the render method of \`` + a + "`.";
        }
        return "";
      }
    }
    function Hn(a) {
      {
        if (a !== void 0) {
          var p = a.fileName.replace(/^.*[\\\/]/, ""), b = a.lineNumber;
          return `

Check your code at ` + p + ":" + b + ".";
        }
        return "";
      }
    }
    var pn = {};
    function zn(a) {
      {
        var p = mn();
        if (!p) {
          var b = typeof a == "string" ? a : a.displayName || a.name;
          b && (p = `

Check the top-level render call using <` + b + ">.");
        }
        return p;
      }
    }
    function vn(a, p) {
      {
        if (!a._store || a._store.validated || a.key != null)
          return;
        a._store.validated = !0;
        var b = zn(p);
        if (pn[b])
          return;
        pn[b] = !0;
        var x = "";
        a && a._owner && a._owner !== Ye.current && (x = " It was passed a child from " + Z(a._owner.type) + "."), Te(a), _('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', b, x), Te(null);
      }
    }
    function hn(a, p) {
      {
        if (typeof a != "object")
          return;
        if (Pe(a))
          for (var b = 0; b < a.length; b++) {
            var x = a[b];
            He(x) && vn(x, p);
          }
        else if (He(a))
          a._store && (a._store.validated = !0);
        else if (a) {
          var j = O(a);
          if (typeof j == "function" && j !== a.entries)
            for (var z = j.call(a), D; !(D = z.next()).done; )
              He(D.value) && vn(D.value, p);
        }
      }
    }
    function Jn(a) {
      {
        var p = a.type;
        if (p == null || typeof p == "string")
          return;
        var b;
        if (typeof p == "function")
          b = p.propTypes;
        else if (typeof p == "object" && (p.$$typeof === c || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        p.$$typeof === v))
          b = p.propTypes;
        else
          return;
        if (b) {
          var x = Z(p);
          sn(b, a.props, "prop", x, a);
        } else if (p.PropTypes !== void 0 && !qe) {
          qe = !0;
          var j = Z(p);
          _("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", j || "Unknown");
        }
        typeof p.getDefaultProps == "function" && !p.getDefaultProps.isReactClassApproved && _("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function Gn(a) {
      {
        for (var p = Object.keys(a.props), b = 0; b < p.length; b++) {
          var x = p[b];
          if (x !== "children" && x !== "key") {
            Te(a), _("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", x), Te(null);
            break;
          }
        }
        a.ref !== null && (Te(a), _("Invalid attribute `ref` supplied to `React.Fragment`."), Te(null));
      }
    }
    function xn(a, p, b, x, j, z) {
      {
        var D = we(a);
        if (!D) {
          var k = "";
          (a === void 0 || typeof a == "object" && a !== null && Object.keys(a).length === 0) && (k += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var ie = Hn(j);
          ie ? k += ie : k += mn();
          var ae;
          a === null ? ae = "null" : Pe(a) ? ae = "array" : a !== void 0 && a.$$typeof === n ? (ae = "<" + (Z(a.type) || "Unknown") + " />", k = " Did you accidentally export a JSX literal instead of a component?") : ae = typeof a, _("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", ae, k);
        }
        var ce = qn(a, p, b, j, z);
        if (ce == null)
          return ce;
        if (D) {
          var me = p.children;
          if (me !== void 0)
            if (x)
              if (Pe(me)) {
                for (var Ze = 0; Ze < me.length; Ze++)
                  hn(me[Ze], a);
                Object.freeze && Object.freeze(me);
              } else
                _("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              hn(me, a);
        }
        return a === t ? Gn(ce) : Jn(ce), ce;
      }
    }
    function Kn(a, p, b) {
      return xn(a, p, b, !0);
    }
    function l(a, p, b) {
      return xn(a, p, b, !1);
    }
    var g = l, R = Kn;
    bn.Fragment = t, bn.jsx = g, bn.jsxs = R;
  }()), bn;
}
var yn = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Yr;
function $o() {
  if (Yr)
    return yn;
  Yr = 1;
  var e = Ee, n = Symbol.for("react.element"), r = Symbol.for("react.fragment"), t = Object.prototype.hasOwnProperty, o = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, i = { key: !0, ref: !0, __self: !0, __source: !0 };
  function u(s, c, f) {
    var m, v = {}, h = null, C = null;
    f !== void 0 && (h = "" + f), c.key !== void 0 && (h = "" + c.key), c.ref !== void 0 && (C = c.ref);
    for (m in c)
      t.call(c, m) && !i.hasOwnProperty(m) && (v[m] = c[m]);
    if (s && s.defaultProps)
      for (m in c = s.defaultProps, c)
        v[m] === void 0 && (v[m] = c[m]);
    return { $$typeof: n, type: s, key: h, ref: C, props: v, _owner: o.current };
  }
  return yn.Fragment = r, yn.jsx = u, yn.jsxs = u, yn;
}
process.env.NODE_ENV === "production" ? mr.exports = $o() : mr.exports = Io();
var P = mr.exports;
const Lo = ["xxl", "xl", "lg", "md", "sm", "xs"], Bo = "xs", On = /* @__PURE__ */ U.createContext({
  prefixes: {},
  breakpoints: Lo,
  minBreakpoint: Bo
});
function de(e, n) {
  const {
    prefixes: r
  } = Ke(On);
  return e || r[n] || n;
}
function jt() {
  const {
    breakpoints: e
  } = Ke(On);
  return e;
}
function _t() {
  const {
    minBreakpoint: e
  } = Ke(On);
  return e;
}
function Wo() {
  const {
    dir: e
  } = Ke(On);
  return e === "rtl";
}
const St = /* @__PURE__ */ U.forwardRef(({
  className: e,
  bsPrefix: n,
  as: r = "div",
  ...t
}, o) => (n = de(n, "card-body"), /* @__PURE__ */ P.jsx(r, {
  ref: o,
  className: re(e, n),
  ...t
})));
St.displayName = "CardBody";
const Tt = St, Ft = /* @__PURE__ */ U.forwardRef(({
  className: e,
  bsPrefix: n,
  as: r = "div",
  ...t
}, o) => (n = de(n, "card-footer"), /* @__PURE__ */ P.jsx(r, {
  ref: o,
  className: re(e, n),
  ...t
})));
Ft.displayName = "CardFooter";
const Uo = Ft, Ot = /* @__PURE__ */ U.createContext(null);
Ot.displayName = "CardHeaderContext";
const Yo = Ot, kt = /* @__PURE__ */ U.forwardRef(({
  bsPrefix: e,
  className: n,
  // Need to define the default "as" during prop destructuring to be compatible with styled-components github.com/react-bootstrap/react-bootstrap/issues/3595
  as: r = "div",
  ...t
}, o) => {
  const i = de(e, "card-header"), u = tn(() => ({
    cardHeaderBsPrefix: i
  }), [i]);
  return /* @__PURE__ */ P.jsx(Yo.Provider, {
    value: u,
    children: /* @__PURE__ */ P.jsx(r, {
      ref: o,
      ...t,
      className: re(n, i)
    })
  });
});
kt.displayName = "CardHeader";
const qo = kt, Dt = /* @__PURE__ */ U.forwardRef(
  // Need to define the default "as" during prop destructuring to be compatible with styled-components github.com/react-bootstrap/react-bootstrap/issues/3595
  ({
    bsPrefix: e,
    className: n,
    variant: r,
    as: t = "img",
    ...o
  }, i) => {
    const u = de(e, "card-img");
    return /* @__PURE__ */ P.jsx(t, {
      ref: i,
      className: re(r ? `${u}-${r}` : u, n),
      ...o
    });
  }
);
Dt.displayName = "CardImg";
const Ho = Dt, At = /* @__PURE__ */ U.forwardRef(({
  className: e,
  bsPrefix: n,
  as: r = "div",
  ...t
}, o) => (n = de(n, "card-img-overlay"), /* @__PURE__ */ P.jsx(r, {
  ref: o,
  className: re(e, n),
  ...t
})));
At.displayName = "CardImgOverlay";
const zo = At, Vt = /* @__PURE__ */ U.forwardRef(({
  className: e,
  bsPrefix: n,
  as: r = "a",
  ...t
}, o) => (n = de(n, "card-link"), /* @__PURE__ */ P.jsx(r, {
  ref: o,
  className: re(e, n),
  ...t
})));
Vt.displayName = "CardLink";
const Jo = Vt, jr = (e) => /* @__PURE__ */ U.forwardRef((n, r) => /* @__PURE__ */ P.jsx("div", {
  ...n,
  ref: r,
  className: re(n.className, e)
})), Go = jr("h6"), Pt = /* @__PURE__ */ U.forwardRef(({
  className: e,
  bsPrefix: n,
  as: r = Go,
  ...t
}, o) => (n = de(n, "card-subtitle"), /* @__PURE__ */ P.jsx(r, {
  ref: o,
  className: re(e, n),
  ...t
})));
Pt.displayName = "CardSubtitle";
const Ko = Pt, Mt = /* @__PURE__ */ U.forwardRef(({
  className: e,
  bsPrefix: n,
  as: r = "p",
  ...t
}, o) => (n = de(n, "card-text"), /* @__PURE__ */ P.jsx(r, {
  ref: o,
  className: re(e, n),
  ...t
})));
Mt.displayName = "CardText";
const Xo = Mt, Zo = jr("h5"), It = /* @__PURE__ */ U.forwardRef(({
  className: e,
  bsPrefix: n,
  as: r = Zo,
  ...t
}, o) => (n = de(n, "card-title"), /* @__PURE__ */ P.jsx(r, {
  ref: o,
  className: re(e, n),
  ...t
})));
It.displayName = "CardTitle";
const Qo = It, $t = /* @__PURE__ */ U.forwardRef(({
  bsPrefix: e,
  className: n,
  bg: r,
  text: t,
  border: o,
  body: i = !1,
  children: u,
  // Need to define the default "as" during prop destructuring to be compatible with styled-components github.com/react-bootstrap/react-bootstrap/issues/3595
  as: s = "div",
  ...c
}, f) => {
  const m = de(e, "card");
  return /* @__PURE__ */ P.jsx(s, {
    ref: f,
    ...c,
    className: re(n, m, r && `bg-${r}`, t && `text-${t}`, o && `border-${o}`),
    children: i ? /* @__PURE__ */ P.jsx(Tt, {
      children: u
    }) : u
  });
});
$t.displayName = "Card";
const Cn = Object.assign($t, {
  Img: Ho,
  Title: Qo,
  Subtitle: Ko,
  Body: Tt,
  Link: Jo,
  Text: Xo,
  Header: qo,
  Footer: Uo,
  ImgOverlay: zo
}), Eu = ({ title: e = "", children: n = null }) => /* @__PURE__ */ d.jsxDEV(Cn, { children: [
  /* @__PURE__ */ d.jsxDEV(Cn.Header, { children: /* @__PURE__ */ d.jsxDEV(Cn.Title, { children: e }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/CardCapsule.jsx",
    lineNumber: 12,
    columnNumber: 13
  }, void 0) }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/CardCapsule.jsx",
    lineNumber: 11,
    columnNumber: 9
  }, void 0),
  /* @__PURE__ */ d.jsxDEV(Cn.Body, { children: n }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/CardCapsule.jsx",
    lineNumber: 16,
    columnNumber: 9
  }, void 0)
] }, void 0, !0, {
  fileName: "/home/runner/work/frontendui/frontendui/src/Components/CardCapsule.jsx",
  lineNumber: 10,
  columnNumber: 5
}, void 0), ei = (e = 300) => {
  let n = -1, r = 0;
  return (t) => {
    r !== 0 && (clearTimeout(n), n = -1, r = 0);
    const o = () => (n = -1, r = 0, t());
    r = 1, n = setTimeout(o, e);
  };
}, xu = ({ children: e, onClick: n }) => {
  const [r, t] = fe(0), o = Ce(() => t(0)), i = Ce(() => t(1));
  return r === 0 ? /* @__PURE__ */ d.jsxDEV("button", { className: "btn btn-sm btn-warning", onClick: i, children: e }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/DeleteButton.jsx",
    lineNumber: 29,
    columnNumber: 13
  }, void 0) : /* @__PURE__ */ d.jsxDEV(d.Fragment, { children: [
    /* @__PURE__ */ d.jsxDEV("button", { className: "btn btn-sm btn-warning", onClick: o, children: e }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/DeleteButton.jsx",
      lineNumber: 35,
      columnNumber: 17
    }, void 0),
    /* @__PURE__ */ d.jsxDEV("button", { className: "btn btn-sm btn-danger", onClick: n, children: e }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/DeleteButton.jsx",
      lineNumber: 36,
      columnNumber: 17
    }, void 0)
  ] }, void 0, !0, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/DeleteButton.jsx",
    lineNumber: 34,
    columnNumber: 13
  }, void 0);
};
var pr = { exports: {} }, Xn, qr;
function _r() {
  if (qr)
    return Xn;
  qr = 1;
  var e = "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED";
  return Xn = e, Xn;
}
var Zn, Hr;
function ni() {
  if (Hr)
    return Zn;
  Hr = 1;
  var e = _r();
  function n() {
  }
  function r() {
  }
  return r.resetWarningCache = n, Zn = function() {
    function t(u, s, c, f, m, v) {
      if (v !== e) {
        var h = new Error(
          "Calling PropTypes validators directly is not supported by the `prop-types` package. Use PropTypes.checkPropTypes() to call them. Read more at http://fb.me/use-check-prop-types"
        );
        throw h.name = "Invariant Violation", h;
      }
    }
    t.isRequired = t;
    function o() {
      return t;
    }
    var i = {
      array: t,
      bigint: t,
      bool: t,
      func: t,
      number: t,
      object: t,
      string: t,
      symbol: t,
      any: t,
      arrayOf: o,
      element: t,
      elementType: t,
      instanceOf: o,
      node: t,
      objectOf: o,
      oneOf: o,
      oneOfType: o,
      shape: o,
      exact: o,
      checkPropTypes: r,
      resetWarningCache: n
    };
    return i.PropTypes = i, i;
  }, Zn;
}
var Rn = { exports: {} }, Q = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var zr;
function ri() {
  return zr || (zr = 1, process.env.NODE_ENV !== "production" && function() {
    var e = typeof Symbol == "function" && Symbol.for, n = e ? Symbol.for("react.element") : 60103, r = e ? Symbol.for("react.portal") : 60106, t = e ? Symbol.for("react.fragment") : 60107, o = e ? Symbol.for("react.strict_mode") : 60108, i = e ? Symbol.for("react.profiler") : 60114, u = e ? Symbol.for("react.provider") : 60109, s = e ? Symbol.for("react.context") : 60110, c = e ? Symbol.for("react.async_mode") : 60111, f = e ? Symbol.for("react.concurrent_mode") : 60111, m = e ? Symbol.for("react.forward_ref") : 60112, v = e ? Symbol.for("react.suspense") : 60113, h = e ? Symbol.for("react.suspense_list") : 60120, C = e ? Symbol.for("react.memo") : 60115, S = e ? Symbol.for("react.lazy") : 60116, T = e ? Symbol.for("react.block") : 60121, O = e ? Symbol.for("react.fundamental") : 60117, A = e ? Symbol.for("react.responder") : 60118, _ = e ? Symbol.for("react.scope") : 60119;
    function B(N) {
      return typeof N == "string" || typeof N == "function" || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
      N === t || N === f || N === i || N === o || N === v || N === h || typeof N == "object" && N !== null && (N.$$typeof === S || N.$$typeof === C || N.$$typeof === u || N.$$typeof === s || N.$$typeof === m || N.$$typeof === O || N.$$typeof === A || N.$$typeof === _ || N.$$typeof === T);
    }
    function q(N) {
      if (typeof N == "object" && N !== null) {
        var ve = N.$$typeof;
        switch (ve) {
          case n:
            var Ne = N.type;
            switch (Ne) {
              case c:
              case f:
              case t:
              case i:
              case o:
              case v:
                return Ne;
              default:
                var K = Ne && Ne.$$typeof;
                switch (K) {
                  case s:
                  case m:
                  case S:
                  case C:
                  case u:
                    return K;
                  default:
                    return ve;
                }
            }
          case r:
            return ve;
        }
      }
    }
    var E = c, G = f, se = s, ye = u, xe = n, we = m, be = t, ge = S, Z = C, te = r, le = i, ne = o, pe = v, oe = !1;
    function Re(N) {
      return oe || (oe = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), y(N) || q(N) === c;
    }
    function y(N) {
      return q(N) === f;
    }
    function w(N) {
      return q(N) === s;
    }
    function M(N) {
      return q(N) === u;
    }
    function V(N) {
      return typeof N == "object" && N !== null && N.$$typeof === n;
    }
    function $(N) {
      return q(N) === m;
    }
    function J(N) {
      return q(N) === t;
    }
    function I(N) {
      return q(N) === S;
    }
    function W(N) {
      return q(N) === C;
    }
    function Y(N) {
      return q(N) === r;
    }
    function H(N) {
      return q(N) === i;
    }
    function L(N) {
      return q(N) === o;
    }
    function ue(N) {
      return q(N) === v;
    }
    Q.AsyncMode = E, Q.ConcurrentMode = G, Q.ContextConsumer = se, Q.ContextProvider = ye, Q.Element = xe, Q.ForwardRef = we, Q.Fragment = be, Q.Lazy = ge, Q.Memo = Z, Q.Portal = te, Q.Profiler = le, Q.StrictMode = ne, Q.Suspense = pe, Q.isAsyncMode = Re, Q.isConcurrentMode = y, Q.isContextConsumer = w, Q.isContextProvider = M, Q.isElement = V, Q.isForwardRef = $, Q.isFragment = J, Q.isLazy = I, Q.isMemo = W, Q.isPortal = Y, Q.isProfiler = H, Q.isStrictMode = L, Q.isSuspense = ue, Q.isValidElementType = B, Q.typeOf = q;
  }()), Q;
}
var ee = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Jr;
function ti() {
  if (Jr)
    return ee;
  Jr = 1;
  var e = typeof Symbol == "function" && Symbol.for, n = e ? Symbol.for("react.element") : 60103, r = e ? Symbol.for("react.portal") : 60106, t = e ? Symbol.for("react.fragment") : 60107, o = e ? Symbol.for("react.strict_mode") : 60108, i = e ? Symbol.for("react.profiler") : 60114, u = e ? Symbol.for("react.provider") : 60109, s = e ? Symbol.for("react.context") : 60110, c = e ? Symbol.for("react.async_mode") : 60111, f = e ? Symbol.for("react.concurrent_mode") : 60111, m = e ? Symbol.for("react.forward_ref") : 60112, v = e ? Symbol.for("react.suspense") : 60113, h = e ? Symbol.for("react.suspense_list") : 60120, C = e ? Symbol.for("react.memo") : 60115, S = e ? Symbol.for("react.lazy") : 60116, T = e ? Symbol.for("react.block") : 60121, O = e ? Symbol.for("react.fundamental") : 60117, A = e ? Symbol.for("react.responder") : 60118, _ = e ? Symbol.for("react.scope") : 60119;
  function B(E) {
    if (typeof E == "object" && E !== null) {
      var G = E.$$typeof;
      switch (G) {
        case n:
          switch (E = E.type, E) {
            case c:
            case f:
            case t:
            case i:
            case o:
            case v:
              return E;
            default:
              switch (E = E && E.$$typeof, E) {
                case s:
                case m:
                case S:
                case C:
                case u:
                  return E;
                default:
                  return G;
              }
          }
        case r:
          return G;
      }
    }
  }
  function q(E) {
    return B(E) === f;
  }
  return ee.AsyncMode = c, ee.ConcurrentMode = f, ee.ContextConsumer = s, ee.ContextProvider = u, ee.Element = n, ee.ForwardRef = m, ee.Fragment = t, ee.Lazy = S, ee.Memo = C, ee.Portal = r, ee.Profiler = i, ee.StrictMode = o, ee.Suspense = v, ee.isAsyncMode = function(E) {
    return q(E) || B(E) === c;
  }, ee.isConcurrentMode = q, ee.isContextConsumer = function(E) {
    return B(E) === s;
  }, ee.isContextProvider = function(E) {
    return B(E) === u;
  }, ee.isElement = function(E) {
    return typeof E == "object" && E !== null && E.$$typeof === n;
  }, ee.isForwardRef = function(E) {
    return B(E) === m;
  }, ee.isFragment = function(E) {
    return B(E) === t;
  }, ee.isLazy = function(E) {
    return B(E) === S;
  }, ee.isMemo = function(E) {
    return B(E) === C;
  }, ee.isPortal = function(E) {
    return B(E) === r;
  }, ee.isProfiler = function(E) {
    return B(E) === i;
  }, ee.isStrictMode = function(E) {
    return B(E) === o;
  }, ee.isSuspense = function(E) {
    return B(E) === v;
  }, ee.isValidElementType = function(E) {
    return typeof E == "string" || typeof E == "function" || E === t || E === f || E === i || E === o || E === v || E === h || typeof E == "object" && E !== null && (E.$$typeof === S || E.$$typeof === C || E.$$typeof === u || E.$$typeof === s || E.$$typeof === m || E.$$typeof === O || E.$$typeof === A || E.$$typeof === _ || E.$$typeof === T);
  }, ee.typeOf = B, ee;
}
var Gr;
function Lt() {
  return Gr || (Gr = 1, process.env.NODE_ENV === "production" ? Rn.exports = ti() : Rn.exports = ri()), Rn.exports;
}
/*
object-assign
(c) Sindre Sorhus
@license MIT
*/
var Qn, Kr;
function oi() {
  if (Kr)
    return Qn;
  Kr = 1;
  var e = Object.getOwnPropertySymbols, n = Object.prototype.hasOwnProperty, r = Object.prototype.propertyIsEnumerable;
  function t(i) {
    if (i == null)
      throw new TypeError("Object.assign cannot be called with null or undefined");
    return Object(i);
  }
  function o() {
    try {
      if (!Object.assign)
        return !1;
      var i = new String("abc");
      if (i[5] = "de", Object.getOwnPropertyNames(i)[0] === "5")
        return !1;
      for (var u = {}, s = 0; s < 10; s++)
        u["_" + String.fromCharCode(s)] = s;
      var c = Object.getOwnPropertyNames(u).map(function(m) {
        return u[m];
      });
      if (c.join("") !== "0123456789")
        return !1;
      var f = {};
      return "abcdefghijklmnopqrst".split("").forEach(function(m) {
        f[m] = m;
      }), Object.keys(Object.assign({}, f)).join("") === "abcdefghijklmnopqrst";
    } catch {
      return !1;
    }
  }
  return Qn = o() ? Object.assign : function(i, u) {
    for (var s, c = t(i), f, m = 1; m < arguments.length; m++) {
      s = Object(arguments[m]);
      for (var v in s)
        n.call(s, v) && (c[v] = s[v]);
      if (e) {
        f = e(s);
        for (var h = 0; h < f.length; h++)
          r.call(s, f[h]) && (c[f[h]] = s[f[h]]);
      }
    }
    return c;
  }, Qn;
}
var er, Xr;
function Bt() {
  return Xr || (Xr = 1, er = Function.call.bind(Object.prototype.hasOwnProperty)), er;
}
var nr, Zr;
function ii() {
  if (Zr)
    return nr;
  Zr = 1;
  var e = function() {
  };
  if (process.env.NODE_ENV !== "production") {
    var n = _r(), r = {}, t = Bt();
    e = function(i) {
      var u = "Warning: " + i;
      typeof console < "u" && console.error(u);
      try {
        throw new Error(u);
      } catch {
      }
    };
  }
  function o(i, u, s, c, f) {
    if (process.env.NODE_ENV !== "production") {
      for (var m in i)
        if (t(i, m)) {
          var v;
          try {
            if (typeof i[m] != "function") {
              var h = Error(
                (c || "React class") + ": " + s + " type `" + m + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof i[m] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`."
              );
              throw h.name = "Invariant Violation", h;
            }
            v = i[m](u, m, c, s, null, n);
          } catch (S) {
            v = S;
          }
          if (v && !(v instanceof Error) && e(
            (c || "React class") + ": type specification of " + s + " `" + m + "` is invalid; the type checker function must return `null` or an `Error` but returned a " + typeof v + ". You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument)."
          ), v instanceof Error && !(v.message in r)) {
            r[v.message] = !0;
            var C = f ? f() : "";
            e(
              "Failed " + s + " type: " + v.message + (C ?? "")
            );
          }
        }
    }
  }
  return o.resetWarningCache = function() {
    process.env.NODE_ENV !== "production" && (r = {});
  }, nr = o, nr;
}
var rr, Qr;
function ai() {
  if (Qr)
    return rr;
  Qr = 1;
  var e = Lt(), n = oi(), r = _r(), t = Bt(), o = ii(), i = function() {
  };
  process.env.NODE_ENV !== "production" && (i = function(s) {
    var c = "Warning: " + s;
    typeof console < "u" && console.error(c);
    try {
      throw new Error(c);
    } catch {
    }
  });
  function u() {
    return null;
  }
  return rr = function(s, c) {
    var f = typeof Symbol == "function" && Symbol.iterator, m = "@@iterator";
    function v(y) {
      var w = y && (f && y[f] || y[m]);
      if (typeof w == "function")
        return w;
    }
    var h = "<<anonymous>>", C = {
      array: A("array"),
      bigint: A("bigint"),
      bool: A("boolean"),
      func: A("function"),
      number: A("number"),
      object: A("object"),
      string: A("string"),
      symbol: A("symbol"),
      any: _(),
      arrayOf: B,
      element: q(),
      elementType: E(),
      instanceOf: G,
      node: we(),
      objectOf: ye,
      oneOf: se,
      oneOfType: xe,
      shape: ge,
      exact: Z
    };
    function S(y, w) {
      return y === w ? y !== 0 || 1 / y === 1 / w : y !== y && w !== w;
    }
    function T(y, w) {
      this.message = y, this.data = w && typeof w == "object" ? w : {}, this.stack = "";
    }
    T.prototype = Error.prototype;
    function O(y) {
      if (process.env.NODE_ENV !== "production")
        var w = {}, M = 0;
      function V(J, I, W, Y, H, L, ue) {
        if (Y = Y || h, L = L || W, ue !== r) {
          if (c) {
            var N = new Error(
              "Calling PropTypes validators directly is not supported by the `prop-types` package. Use `PropTypes.checkPropTypes()` to call them. Read more at http://fb.me/use-check-prop-types"
            );
            throw N.name = "Invariant Violation", N;
          } else if (process.env.NODE_ENV !== "production" && typeof console < "u") {
            var ve = Y + ":" + W;
            !w[ve] && // Avoid spamming the console because they are often not actionable except for lib authors
            M < 3 && (i(
              "You are manually calling a React.PropTypes validation function for the `" + L + "` prop on `" + Y + "`. This is deprecated and will throw in the standalone `prop-types` package. You may be seeing this warning due to a third-party PropTypes library. See https://fb.me/react-warning-dont-call-proptypes for details."
            ), w[ve] = !0, M++);
          }
        }
        return I[W] == null ? J ? I[W] === null ? new T("The " + H + " `" + L + "` is marked as required " + ("in `" + Y + "`, but its value is `null`.")) : new T("The " + H + " `" + L + "` is marked as required in " + ("`" + Y + "`, but its value is `undefined`.")) : null : y(I, W, Y, H, L);
      }
      var $ = V.bind(null, !1);
      return $.isRequired = V.bind(null, !0), $;
    }
    function A(y) {
      function w(M, V, $, J, I, W) {
        var Y = M[V], H = ne(Y);
        if (H !== y) {
          var L = pe(Y);
          return new T(
            "Invalid " + J + " `" + I + "` of type " + ("`" + L + "` supplied to `" + $ + "`, expected ") + ("`" + y + "`."),
            { expectedType: y }
          );
        }
        return null;
      }
      return O(w);
    }
    function _() {
      return O(u);
    }
    function B(y) {
      function w(M, V, $, J, I) {
        if (typeof y != "function")
          return new T("Property `" + I + "` of component `" + $ + "` has invalid PropType notation inside arrayOf.");
        var W = M[V];
        if (!Array.isArray(W)) {
          var Y = ne(W);
          return new T("Invalid " + J + " `" + I + "` of type " + ("`" + Y + "` supplied to `" + $ + "`, expected an array."));
        }
        for (var H = 0; H < W.length; H++) {
          var L = y(W, H, $, J, I + "[" + H + "]", r);
          if (L instanceof Error)
            return L;
        }
        return null;
      }
      return O(w);
    }
    function q() {
      function y(w, M, V, $, J) {
        var I = w[M];
        if (!s(I)) {
          var W = ne(I);
          return new T("Invalid " + $ + " `" + J + "` of type " + ("`" + W + "` supplied to `" + V + "`, expected a single ReactElement."));
        }
        return null;
      }
      return O(y);
    }
    function E() {
      function y(w, M, V, $, J) {
        var I = w[M];
        if (!e.isValidElementType(I)) {
          var W = ne(I);
          return new T("Invalid " + $ + " `" + J + "` of type " + ("`" + W + "` supplied to `" + V + "`, expected a single ReactElement type."));
        }
        return null;
      }
      return O(y);
    }
    function G(y) {
      function w(M, V, $, J, I) {
        if (!(M[V] instanceof y)) {
          var W = y.name || h, Y = Re(M[V]);
          return new T("Invalid " + J + " `" + I + "` of type " + ("`" + Y + "` supplied to `" + $ + "`, expected ") + ("instance of `" + W + "`."));
        }
        return null;
      }
      return O(w);
    }
    function se(y) {
      if (!Array.isArray(y))
        return process.env.NODE_ENV !== "production" && (arguments.length > 1 ? i(
          "Invalid arguments supplied to oneOf, expected an array, got " + arguments.length + " arguments. A common mistake is to write oneOf(x, y, z) instead of oneOf([x, y, z])."
        ) : i("Invalid argument supplied to oneOf, expected an array.")), u;
      function w(M, V, $, J, I) {
        for (var W = M[V], Y = 0; Y < y.length; Y++)
          if (S(W, y[Y]))
            return null;
        var H = JSON.stringify(y, function(ue, N) {
          var ve = pe(N);
          return ve === "symbol" ? String(N) : N;
        });
        return new T("Invalid " + J + " `" + I + "` of value `" + String(W) + "` " + ("supplied to `" + $ + "`, expected one of " + H + "."));
      }
      return O(w);
    }
    function ye(y) {
      function w(M, V, $, J, I) {
        if (typeof y != "function")
          return new T("Property `" + I + "` of component `" + $ + "` has invalid PropType notation inside objectOf.");
        var W = M[V], Y = ne(W);
        if (Y !== "object")
          return new T("Invalid " + J + " `" + I + "` of type " + ("`" + Y + "` supplied to `" + $ + "`, expected an object."));
        for (var H in W)
          if (t(W, H)) {
            var L = y(W, H, $, J, I + "." + H, r);
            if (L instanceof Error)
              return L;
          }
        return null;
      }
      return O(w);
    }
    function xe(y) {
      if (!Array.isArray(y))
        return process.env.NODE_ENV !== "production" && i("Invalid argument supplied to oneOfType, expected an instance of array."), u;
      for (var w = 0; w < y.length; w++) {
        var M = y[w];
        if (typeof M != "function")
          return i(
            "Invalid argument supplied to oneOfType. Expected an array of check functions, but received " + oe(M) + " at index " + w + "."
          ), u;
      }
      function V($, J, I, W, Y) {
        for (var H = [], L = 0; L < y.length; L++) {
          var ue = y[L], N = ue($, J, I, W, Y, r);
          if (N == null)
            return null;
          N.data && t(N.data, "expectedType") && H.push(N.data.expectedType);
        }
        var ve = H.length > 0 ? ", expected one of type [" + H.join(", ") + "]" : "";
        return new T("Invalid " + W + " `" + Y + "` supplied to " + ("`" + I + "`" + ve + "."));
      }
      return O(V);
    }
    function we() {
      function y(w, M, V, $, J) {
        return te(w[M]) ? null : new T("Invalid " + $ + " `" + J + "` supplied to " + ("`" + V + "`, expected a ReactNode."));
      }
      return O(y);
    }
    function be(y, w, M, V, $) {
      return new T(
        (y || "React class") + ": " + w + " type `" + M + "." + V + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + $ + "`."
      );
    }
    function ge(y) {
      function w(M, V, $, J, I) {
        var W = M[V], Y = ne(W);
        if (Y !== "object")
          return new T("Invalid " + J + " `" + I + "` of type `" + Y + "` " + ("supplied to `" + $ + "`, expected `object`."));
        for (var H in y) {
          var L = y[H];
          if (typeof L != "function")
            return be($, J, I, H, pe(L));
          var ue = L(W, H, $, J, I + "." + H, r);
          if (ue)
            return ue;
        }
        return null;
      }
      return O(w);
    }
    function Z(y) {
      function w(M, V, $, J, I) {
        var W = M[V], Y = ne(W);
        if (Y !== "object")
          return new T("Invalid " + J + " `" + I + "` of type `" + Y + "` " + ("supplied to `" + $ + "`, expected `object`."));
        var H = n({}, M[V], y);
        for (var L in H) {
          var ue = y[L];
          if (t(y, L) && typeof ue != "function")
            return be($, J, I, L, pe(ue));
          if (!ue)
            return new T(
              "Invalid " + J + " `" + I + "` key `" + L + "` supplied to `" + $ + "`.\nBad object: " + JSON.stringify(M[V], null, "  ") + `
Valid keys: ` + JSON.stringify(Object.keys(y), null, "  ")
            );
          var N = ue(W, L, $, J, I + "." + L, r);
          if (N)
            return N;
        }
        return null;
      }
      return O(w);
    }
    function te(y) {
      switch (typeof y) {
        case "number":
        case "string":
        case "undefined":
          return !0;
        case "boolean":
          return !y;
        case "object":
          if (Array.isArray(y))
            return y.every(te);
          if (y === null || s(y))
            return !0;
          var w = v(y);
          if (w) {
            var M = w.call(y), V;
            if (w !== y.entries) {
              for (; !(V = M.next()).done; )
                if (!te(V.value))
                  return !1;
            } else
              for (; !(V = M.next()).done; ) {
                var $ = V.value;
                if ($ && !te($[1]))
                  return !1;
              }
          } else
            return !1;
          return !0;
        default:
          return !1;
      }
    }
    function le(y, w) {
      return y === "symbol" ? !0 : w ? w["@@toStringTag"] === "Symbol" || typeof Symbol == "function" && w instanceof Symbol : !1;
    }
    function ne(y) {
      var w = typeof y;
      return Array.isArray(y) ? "array" : y instanceof RegExp ? "object" : le(w, y) ? "symbol" : w;
    }
    function pe(y) {
      if (typeof y > "u" || y === null)
        return "" + y;
      var w = ne(y);
      if (w === "object") {
        if (y instanceof Date)
          return "date";
        if (y instanceof RegExp)
          return "regexp";
      }
      return w;
    }
    function oe(y) {
      var w = pe(y);
      switch (w) {
        case "array":
        case "object":
          return "an " + w;
        case "boolean":
        case "date":
        case "regexp":
          return "a " + w;
        default:
          return w;
      }
    }
    function Re(y) {
      return !y.constructor || !y.constructor.name ? h : y.constructor.name;
    }
    return C.checkPropTypes = o, C.resetWarningCache = o.resetWarningCache, C.PropTypes = C, C;
  }, rr;
}
if (process.env.NODE_ENV !== "production") {
  var si = Lt(), ui = !0;
  pr.exports = ai()(si.isElement, ui);
} else
  pr.exports = ni()();
var li = pr.exports;
const F = /* @__PURE__ */ Ct(li), ci = ["as", "disabled"];
function fi(e, n) {
  if (e == null)
    return {};
  var r = {}, t = Object.keys(e), o, i;
  for (i = 0; i < t.length; i++)
    o = t[i], !(n.indexOf(o) >= 0) && (r[o] = e[o]);
  return r;
}
function di(e) {
  return !e || e.trim() === "#";
}
function Wt({
  tagName: e,
  disabled: n,
  href: r,
  target: t,
  rel: o,
  role: i,
  onClick: u,
  tabIndex: s = 0,
  type: c
}) {
  e || (r != null || t != null || o != null ? e = "a" : e = "button");
  const f = {
    tagName: e
  };
  if (e === "button")
    return [{
      type: c || "button",
      disabled: n
    }, f];
  const m = (h) => {
    if ((n || e === "a" && di(r)) && h.preventDefault(), n) {
      h.stopPropagation();
      return;
    }
    u == null || u(h);
  }, v = (h) => {
    h.key === " " && (h.preventDefault(), m(h));
  };
  return e === "a" && (r || (r = "#"), n && (r = void 0)), [{
    role: i ?? "button",
    // explicitly undefined so that it overrides the props disabled in a spread
    // e.g. <Tag {...props} {...hookProps} />
    disabled: void 0,
    tabIndex: n ? void 0 : s,
    href: r,
    target: e === "a" ? t : void 0,
    "aria-disabled": n || void 0,
    rel: e === "a" ? o : void 0,
    onClick: m,
    onKeyDown: v
  }, f];
}
const mi = /* @__PURE__ */ U.forwardRef((e, n) => {
  let {
    as: r,
    disabled: t
  } = e, o = fi(e, ci);
  const [i, {
    tagName: u
  }] = Wt(Object.assign({
    tagName: r,
    disabled: t
  }, o));
  return /* @__PURE__ */ P.jsx(u, Object.assign({}, o, i, {
    ref: n
  }));
});
mi.displayName = "Button";
const Ut = /* @__PURE__ */ U.forwardRef(({
  as: e,
  bsPrefix: n,
  variant: r = "primary",
  size: t,
  active: o = !1,
  disabled: i = !1,
  className: u,
  ...s
}, c) => {
  const f = de(n, "btn"), [m, {
    tagName: v
  }] = Wt({
    tagName: e,
    disabled: i,
    ...s
  }), h = v;
  return /* @__PURE__ */ P.jsx(h, {
    ...m,
    ...s,
    ref: c,
    disabled: i,
    className: re(u, f, o && "active", r && `${f}-${r}`, t && `${f}-${t}`, s.href && i && "disabled")
  });
});
Ut.displayName = "Button";
const et = Ut, on = !!(typeof window < "u" && window.document && window.document.createElement);
var vr = !1, hr = !1;
try {
  var tr = {
    get passive() {
      return vr = !0;
    },
    get once() {
      return hr = vr = !0;
    }
  };
  on && (window.addEventListener("test", tr, tr), window.removeEventListener("test", tr, !0));
} catch {
}
function Yt(e, n, r, t) {
  if (t && typeof t != "boolean" && !hr) {
    var o = t.once, i = t.capture, u = r;
    !hr && o && (u = r.__once || function s(c) {
      this.removeEventListener(n, s, i), r.call(this, c);
    }, r.__once = u), e.addEventListener(n, u, vr ? t : i);
  }
  e.addEventListener(n, r, t);
}
function kn(e) {
  return e && e.ownerDocument || document;
}
function br(e, n, r, t) {
  var o = t && typeof t != "boolean" ? t.capture : t;
  e.removeEventListener(n, r, o), r.__once && e.removeEventListener(n, r.__once, o);
}
var jn;
function nt(e) {
  if ((!jn && jn !== 0 || e) && on) {
    var n = document.createElement("div");
    n.style.position = "absolute", n.style.top = "-9999px", n.style.width = "50px", n.style.height = "50px", n.style.overflow = "scroll", document.body.appendChild(n), jn = n.offsetWidth - n.clientWidth, document.body.removeChild(n);
  }
  return jn;
}
function pi() {
  return fe(null);
}
function vi(e) {
  const n = he(e);
  return Se(() => {
    n.current = e;
  }, [e]), n;
}
function Ae(e) {
  const n = vi(e);
  return Ce(function(...r) {
    return n.current && n.current(...r);
  }, [n]);
}
const rt = (e) => !e || typeof e == "function" ? e : (n) => {
  e.current = n;
};
function hi(e, n) {
  const r = rt(e), t = rt(n);
  return (o) => {
    r && r(o), t && t(o);
  };
}
function Dn(e, n) {
  return tn(() => hi(e, n), [e, n]);
}
function bi(e) {
  const n = he(e);
  return n.current = e, n;
}
function Sr(e) {
  const n = bi(e);
  Se(() => () => n.current(), []);
}
function yi(e) {
  var n = kn(e);
  return n && n.defaultView || window;
}
function gi(e, n) {
  return yi(e).getComputedStyle(e, n);
}
var Ei = /([A-Z])/g;
function xi(e) {
  return e.replace(Ei, "-$1").toLowerCase();
}
var Ni = /^ms-/;
function _n(e) {
  return xi(e).replace(Ni, "-ms-");
}
var wi = /^((translate|rotate|scale)(X|Y|Z|3d)?|matrix(3d)?|perspective|skew(X|Y)?)$/i;
function Ci(e) {
  return !!(e && wi.test(e));
}
function Ge(e, n) {
  var r = "", t = "";
  if (typeof n == "string")
    return e.style.getPropertyValue(_n(n)) || gi(e).getPropertyValue(_n(n));
  Object.keys(n).forEach(function(o) {
    var i = n[o];
    !i && i !== 0 ? e.style.removeProperty(_n(o)) : Ci(o) ? t += o + "(" + i + ") " : r += _n(o) + ": " + i + ";";
  }), t && (r += "transform: " + t + ";"), e.style.cssText += ";" + r;
}
function Sn(e, n, r, t) {
  return Yt(e, n, r, t), function() {
    br(e, n, r, t);
  };
}
function Ri(e, n, r, t) {
  if (r === void 0 && (r = !1), t === void 0 && (t = !0), e) {
    var o = document.createEvent("HTMLEvents");
    o.initEvent(n, r, t), e.dispatchEvent(o);
  }
}
function ji(e) {
  var n = Ge(e, "transitionDuration") || "", r = n.indexOf("ms") === -1 ? 1e3 : 1;
  return parseFloat(n) * r;
}
function _i(e, n, r) {
  r === void 0 && (r = 5);
  var t = !1, o = setTimeout(function() {
    t || Ri(e, "transitionend", !0);
  }, n + r), i = Sn(e, "transitionend", function() {
    t = !0;
  }, {
    once: !0
  });
  return function() {
    clearTimeout(o), i();
  };
}
function qt(e, n, r, t) {
  r == null && (r = ji(e) || 0);
  var o = _i(e, r, t), i = Sn(e, "transitionend", n);
  return function() {
    o(), i();
  };
}
function or(e) {
  e === void 0 && (e = kn());
  try {
    var n = e.activeElement;
    return !n || !n.nodeName ? null : n;
  } catch {
    return e.body;
  }
}
function tt(e, n) {
  if (e.contains)
    return e.contains(n);
  if (e.compareDocumentPosition)
    return e === n || !!(e.compareDocumentPosition(n) & 16);
}
function Ht() {
  const e = he(!0), n = he(() => e.current);
  return Se(() => (e.current = !0, () => {
    e.current = !1;
  }), []), n.current;
}
function Si(e) {
  const n = he(null);
  return Se(() => {
    n.current = e;
  }), n.current;
}
const Ti = "data-rr-ui-";
function Fi(e) {
  return `${Ti}${e}`;
}
function Oi(e = document) {
  const n = e.defaultView;
  return Math.abs(n.innerWidth - e.documentElement.clientWidth);
}
const ot = Fi("modal-open");
class ki {
  constructor({
    ownerDocument: n,
    handleContainerOverflow: r = !0,
    isRTL: t = !1
  } = {}) {
    this.handleContainerOverflow = r, this.isRTL = t, this.modals = [], this.ownerDocument = n;
  }
  getScrollbarWidth() {
    return Oi(this.ownerDocument);
  }
  getElement() {
    return (this.ownerDocument || document).body;
  }
  setModalAttributes(n) {
  }
  removeModalAttributes(n) {
  }
  setContainerStyle(n) {
    const r = {
      overflow: "hidden"
    }, t = this.isRTL ? "paddingLeft" : "paddingRight", o = this.getElement();
    n.style = {
      overflow: o.style.overflow,
      [t]: o.style[t]
    }, n.scrollBarWidth && (r[t] = `${parseInt(Ge(o, t) || "0", 10) + n.scrollBarWidth}px`), o.setAttribute(ot, ""), Ge(o, r);
  }
  reset() {
    [...this.modals].forEach((n) => this.remove(n));
  }
  removeContainerStyle(n) {
    const r = this.getElement();
    r.removeAttribute(ot), Object.assign(r.style, n.style);
  }
  add(n) {
    let r = this.modals.indexOf(n);
    return r !== -1 || (r = this.modals.length, this.modals.push(n), this.setModalAttributes(n), r !== 0) || (this.state = {
      scrollBarWidth: this.getScrollbarWidth(),
      style: {}
    }, this.handleContainerOverflow && this.setContainerStyle(this.state)), r;
  }
  remove(n) {
    const r = this.modals.indexOf(n);
    r !== -1 && (this.modals.splice(r, 1), !this.modals.length && this.handleContainerOverflow && this.removeContainerStyle(this.state), this.removeModalAttributes(n));
  }
  isTopModal(n) {
    return !!this.modals.length && this.modals[this.modals.length - 1] === n;
  }
}
const Tr = ki, zt = /* @__PURE__ */ Oo(on ? window : void 0);
zt.Provider;
function Fr() {
  return Ke(zt);
}
const ir = (e, n) => on ? e == null ? (n || kn()).body : (typeof e == "function" && (e = e()), e && "current" in e && (e = e.current), e && ("nodeType" in e || e.getBoundingClientRect) ? e : null) : null;
function Di(e, n) {
  const r = Fr(), [t, o] = fe(() => ir(e, r == null ? void 0 : r.document));
  if (!t) {
    const i = ir(e);
    i && o(i);
  }
  return Se(() => {
    n && t && n(t);
  }, [n, t]), Se(() => {
    const i = ir(e);
    i !== t && o(i);
  }, [e, t]), t;
}
const Ai = typeof global < "u" && // @ts-ignore
global.navigator && // @ts-ignore
global.navigator.product === "ReactNative", Vi = typeof document < "u", it = Vi || Ai ? ko : Se;
function Pi({
  children: e,
  in: n,
  onExited: r,
  mountOnEnter: t,
  unmountOnExit: o
}) {
  const i = he(null), u = he(n), s = Ae(r);
  Se(() => {
    n ? u.current = !0 : s(i.current);
  }, [n, s]);
  const c = Dn(i, e.ref), f = /* @__PURE__ */ Nt(e, {
    ref: c
  });
  return n ? f : o || !u.current && t ? null : f;
}
function Mi({
  in: e,
  onTransition: n
}) {
  const r = he(null), t = he(!0), o = Ae(n);
  return it(() => {
    if (!r.current)
      return;
    let i = !1;
    return o({
      in: e,
      element: r.current,
      initial: t.current,
      isStale: () => i
    }), () => {
      i = !0;
    };
  }, [e, o]), it(() => (t.current = !1, () => {
    t.current = !0;
  }), []), r;
}
function Ii({
  children: e,
  in: n,
  onExited: r,
  onEntered: t,
  transition: o
}) {
  const [i, u] = fe(!n);
  n && i && u(!1);
  const s = Mi({
    in: !!n,
    onTransition: (f) => {
      const m = () => {
        f.isStale() || (f.in ? t == null || t(f.element, f.initial) : (u(!0), r == null || r(f.element)));
      };
      Promise.resolve(o(f)).then(m, (v) => {
        throw f.in || u(!0), v;
      });
    }
  }), c = Dn(s, e.ref);
  return i && !n ? null : /* @__PURE__ */ Nt(e, {
    ref: c
  });
}
function at(e, n, r) {
  return e ? /* @__PURE__ */ P.jsx(e, Object.assign({}, r)) : n ? /* @__PURE__ */ P.jsx(Ii, Object.assign({}, r, {
    transition: n
  })) : /* @__PURE__ */ P.jsx(Pi, Object.assign({}, r));
}
function $i(e) {
  return e.code === "Escape" || e.keyCode === 27;
}
const Li = ["show", "role", "className", "style", "children", "backdrop", "keyboard", "onBackdropClick", "onEscapeKeyDown", "transition", "runTransition", "backdropTransition", "runBackdropTransition", "autoFocus", "enforceFocus", "restoreFocus", "restoreFocusOptions", "renderDialog", "renderBackdrop", "manager", "container", "onShow", "onHide", "onExit", "onExited", "onExiting", "onEnter", "onEntering", "onEntered"];
function Bi(e, n) {
  if (e == null)
    return {};
  var r = {}, t = Object.keys(e), o, i;
  for (i = 0; i < t.length; i++)
    o = t[i], !(n.indexOf(o) >= 0) && (r[o] = e[o]);
  return r;
}
let ar;
function Wi(e) {
  return ar || (ar = new Tr({
    ownerDocument: e == null ? void 0 : e.document
  })), ar;
}
function Ui(e) {
  const n = Fr(), r = e || Wi(n), t = he({
    dialog: null,
    backdrop: null
  });
  return Object.assign(t.current, {
    add: () => r.add(t.current),
    remove: () => r.remove(t.current),
    isTopModal: () => r.isTopModal(t.current),
    setDialogRef: Ce((o) => {
      t.current.dialog = o;
    }, []),
    setBackdropRef: Ce((o) => {
      t.current.backdrop = o;
    }, [])
  });
}
const Jt = /* @__PURE__ */ Rr((e, n) => {
  let {
    show: r = !1,
    role: t = "dialog",
    className: o,
    style: i,
    children: u,
    backdrop: s = !0,
    keyboard: c = !0,
    onBackdropClick: f,
    onEscapeKeyDown: m,
    transition: v,
    runTransition: h,
    backdropTransition: C,
    runBackdropTransition: S,
    autoFocus: T = !0,
    enforceFocus: O = !0,
    restoreFocus: A = !0,
    restoreFocusOptions: _,
    renderDialog: B,
    renderBackdrop: q = (K) => /* @__PURE__ */ P.jsx("div", Object.assign({}, K)),
    manager: E,
    container: G,
    onShow: se,
    onHide: ye = () => {
    },
    onExit: xe,
    onExited: we,
    onExiting: be,
    onEnter: ge,
    onEntering: Z,
    onEntered: te
  } = e, le = Bi(e, Li);
  const ne = Fr(), pe = Di(G), oe = Ui(E), Re = Ht(), y = Si(r), [w, M] = fe(!r), V = he(null);
  Do(n, () => oe, [oe]), on && !y && r && (V.current = or(ne == null ? void 0 : ne.document)), r && w && M(!1);
  const $ = Ae(() => {
    if (oe.add(), L.current = Sn(document, "keydown", Y), H.current = Sn(
      document,
      "focus",
      // the timeout is necessary b/c this will run before the new modal is mounted
      // and so steals focus from it
      () => setTimeout(I),
      !0
    ), se && se(), T) {
      var K, je;
      const ke = or((K = (je = oe.dialog) == null ? void 0 : je.ownerDocument) != null ? K : ne == null ? void 0 : ne.document);
      oe.dialog && ke && !tt(oe.dialog, ke) && (V.current = ke, oe.dialog.focus());
    }
  }), J = Ae(() => {
    if (oe.remove(), L.current == null || L.current(), H.current == null || H.current(), A) {
      var K;
      (K = V.current) == null || K.focus == null || K.focus(_), V.current = null;
    }
  });
  Se(() => {
    !r || !pe || $();
  }, [
    r,
    pe,
    /* should never change: */
    $
  ]), Se(() => {
    w && J();
  }, [w, J]), Sr(() => {
    J();
  });
  const I = Ae(() => {
    if (!O || !Re() || !oe.isTopModal())
      return;
    const K = or(ne == null ? void 0 : ne.document);
    oe.dialog && K && !tt(oe.dialog, K) && oe.dialog.focus();
  }), W = Ae((K) => {
    K.target === K.currentTarget && (f == null || f(K), s === !0 && ye());
  }), Y = Ae((K) => {
    c && $i(K) && oe.isTopModal() && (m == null || m(K), K.defaultPrevented || ye());
  }), H = he(), L = he(), ue = (...K) => {
    M(!0), we == null || we(...K);
  };
  if (!pe)
    return null;
  const N = Object.assign({
    role: t,
    ref: oe.setDialogRef,
    // apparently only works on the dialog role element
    "aria-modal": t === "dialog" ? !0 : void 0
  }, le, {
    style: i,
    className: o,
    tabIndex: -1
  });
  let ve = B ? B(N) : /* @__PURE__ */ P.jsx("div", Object.assign({}, N, {
    children: /* @__PURE__ */ U.cloneElement(u, {
      role: "document"
    })
  }));
  ve = at(v, h, {
    unmountOnExit: !0,
    mountOnEnter: !0,
    appear: !0,
    in: !!r,
    onExit: xe,
    onExiting: be,
    onExited: ue,
    onEnter: ge,
    onEntering: Z,
    onEntered: te,
    children: ve
  });
  let Ne = null;
  return s && (Ne = q({
    ref: oe.setBackdropRef,
    onClick: W
  }), Ne = at(C, S, {
    in: !!r,
    appear: !0,
    mountOnEnter: !0,
    unmountOnExit: !0,
    children: Ne
  })), /* @__PURE__ */ P.jsx(P.Fragment, {
    children: /* @__PURE__ */ rn.createPortal(/* @__PURE__ */ P.jsxs(P.Fragment, {
      children: [Ne, ve]
    }), pe)
  });
});
Jt.displayName = "Modal";
const Yi = Object.assign(Jt, {
  Manager: Tr
});
function qi(e, n) {
  return e.classList ? !!n && e.classList.contains(n) : (" " + (e.className.baseVal || e.className) + " ").indexOf(" " + n + " ") !== -1;
}
function Hi(e, n) {
  e.classList ? e.classList.add(n) : qi(e, n) || (typeof e.className == "string" ? e.className = e.className + " " + n : e.setAttribute("class", (e.className && e.className.baseVal || "") + " " + n));
}
var zi = Function.prototype.bind.call(Function.prototype.call, [].slice);
function Qe(e, n) {
  return zi(e.querySelectorAll(n));
}
function st(e, n) {
  return e.replace(new RegExp("(^|\\s)" + n + "(?:\\s|$)", "g"), "$1").replace(/\s+/g, " ").replace(/^\s*|\s*$/g, "");
}
function Ji(e, n) {
  e.classList ? e.classList.remove(n) : typeof e.className == "string" ? e.className = st(e.className, n) : e.setAttribute("class", st(e.className && e.className.baseVal || "", n));
}
const en = {
  FIXED_CONTENT: ".fixed-top, .fixed-bottom, .is-fixed, .sticky-top",
  STICKY_CONTENT: ".sticky-top",
  NAVBAR_TOGGLER: ".navbar-toggler"
};
class Gi extends Tr {
  adjustAndStore(n, r, t) {
    const o = r.style[n];
    r.dataset[n] = o, Ge(r, {
      [n]: `${parseFloat(Ge(r, n)) + t}px`
    });
  }
  restore(n, r) {
    const t = r.dataset[n];
    t !== void 0 && (delete r.dataset[n], Ge(r, {
      [n]: t
    }));
  }
  setContainerStyle(n) {
    super.setContainerStyle(n);
    const r = this.getElement();
    if (Hi(r, "modal-open"), !n.scrollBarWidth)
      return;
    const t = this.isRTL ? "paddingLeft" : "paddingRight", o = this.isRTL ? "marginLeft" : "marginRight";
    Qe(r, en.FIXED_CONTENT).forEach((i) => this.adjustAndStore(t, i, n.scrollBarWidth)), Qe(r, en.STICKY_CONTENT).forEach((i) => this.adjustAndStore(o, i, -n.scrollBarWidth)), Qe(r, en.NAVBAR_TOGGLER).forEach((i) => this.adjustAndStore(o, i, n.scrollBarWidth));
  }
  removeContainerStyle(n) {
    super.removeContainerStyle(n);
    const r = this.getElement();
    Ji(r, "modal-open");
    const t = this.isRTL ? "paddingLeft" : "paddingRight", o = this.isRTL ? "marginLeft" : "marginRight";
    Qe(r, en.FIXED_CONTENT).forEach((i) => this.restore(t, i)), Qe(r, en.STICKY_CONTENT).forEach((i) => this.restore(o, i)), Qe(r, en.NAVBAR_TOGGLER).forEach((i) => this.restore(o, i));
  }
}
let sr;
function Ki(e) {
  return sr || (sr = new Gi(e)), sr;
}
function Xi(e, n) {
  if (e == null)
    return {};
  var r = {}, t = Object.keys(e), o, i;
  for (i = 0; i < t.length; i++)
    o = t[i], !(n.indexOf(o) >= 0) && (r[o] = e[o]);
  return r;
}
function yr(e, n) {
  return yr = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(t, o) {
    return t.__proto__ = o, t;
  }, yr(e, n);
}
function Zi(e, n) {
  e.prototype = Object.create(n.prototype), e.prototype.constructor = e, yr(e, n);
}
const ut = {
  disabled: !1
};
var Qi = process.env.NODE_ENV !== "production" ? F.oneOfType([F.number, F.shape({
  enter: F.number,
  exit: F.number,
  appear: F.number
}).isRequired]) : null;
process.env.NODE_ENV !== "production" && F.oneOfType([F.string, F.shape({
  enter: F.string,
  exit: F.string,
  active: F.string
}), F.shape({
  enter: F.string,
  enterDone: F.string,
  enterActive: F.string,
  exit: F.string,
  exitDone: F.string,
  exitActive: F.string
})]);
const Gt = Ee.createContext(null);
var ea = function(n) {
  return n.scrollTop;
}, En = "unmounted", ze = "exited", Ie = "entering", Je = "entered", Tn = "exiting", Le = /* @__PURE__ */ function(e) {
  Zi(n, e);
  function n(t, o) {
    var i;
    i = e.call(this, t, o) || this;
    var u = o, s = u && !u.isMounting ? t.enter : t.appear, c;
    return i.appearStatus = null, t.in ? s ? (c = ze, i.appearStatus = Ie) : c = Je : t.unmountOnExit || t.mountOnEnter ? c = En : c = ze, i.state = {
      status: c
    }, i.nextCallback = null, i;
  }
  n.getDerivedStateFromProps = function(o, i) {
    var u = o.in;
    return u && i.status === En ? {
      status: ze
    } : null;
  };
  var r = n.prototype;
  return r.componentDidMount = function() {
    this.updateStatus(!0, this.appearStatus);
  }, r.componentDidUpdate = function(o) {
    var i = null;
    if (o !== this.props) {
      var u = this.state.status;
      this.props.in ? u !== Ie && u !== Je && (i = Ie) : (u === Ie || u === Je) && (i = Tn);
    }
    this.updateStatus(!1, i);
  }, r.componentWillUnmount = function() {
    this.cancelNextCallback();
  }, r.getTimeouts = function() {
    var o = this.props.timeout, i, u, s;
    return i = u = s = o, o != null && typeof o != "number" && (i = o.exit, u = o.enter, s = o.appear !== void 0 ? o.appear : u), {
      exit: i,
      enter: u,
      appear: s
    };
  }, r.updateStatus = function(o, i) {
    if (o === void 0 && (o = !1), i !== null)
      if (this.cancelNextCallback(), i === Ie) {
        if (this.props.unmountOnExit || this.props.mountOnEnter) {
          var u = this.props.nodeRef ? this.props.nodeRef.current : rn.findDOMNode(this);
          u && ea(u);
        }
        this.performEnter(o);
      } else
        this.performExit();
    else
      this.props.unmountOnExit && this.state.status === ze && this.setState({
        status: En
      });
  }, r.performEnter = function(o) {
    var i = this, u = this.props.enter, s = this.context ? this.context.isMounting : o, c = this.props.nodeRef ? [s] : [rn.findDOMNode(this), s], f = c[0], m = c[1], v = this.getTimeouts(), h = s ? v.appear : v.enter;
    if (!o && !u || ut.disabled) {
      this.safeSetState({
        status: Je
      }, function() {
        i.props.onEntered(f);
      });
      return;
    }
    this.props.onEnter(f, m), this.safeSetState({
      status: Ie
    }, function() {
      i.props.onEntering(f, m), i.onTransitionEnd(h, function() {
        i.safeSetState({
          status: Je
        }, function() {
          i.props.onEntered(f, m);
        });
      });
    });
  }, r.performExit = function() {
    var o = this, i = this.props.exit, u = this.getTimeouts(), s = this.props.nodeRef ? void 0 : rn.findDOMNode(this);
    if (!i || ut.disabled) {
      this.safeSetState({
        status: ze
      }, function() {
        o.props.onExited(s);
      });
      return;
    }
    this.props.onExit(s), this.safeSetState({
      status: Tn
    }, function() {
      o.props.onExiting(s), o.onTransitionEnd(u.exit, function() {
        o.safeSetState({
          status: ze
        }, function() {
          o.props.onExited(s);
        });
      });
    });
  }, r.cancelNextCallback = function() {
    this.nextCallback !== null && (this.nextCallback.cancel(), this.nextCallback = null);
  }, r.safeSetState = function(o, i) {
    i = this.setNextCallback(i), this.setState(o, i);
  }, r.setNextCallback = function(o) {
    var i = this, u = !0;
    return this.nextCallback = function(s) {
      u && (u = !1, i.nextCallback = null, o(s));
    }, this.nextCallback.cancel = function() {
      u = !1;
    }, this.nextCallback;
  }, r.onTransitionEnd = function(o, i) {
    this.setNextCallback(i);
    var u = this.props.nodeRef ? this.props.nodeRef.current : rn.findDOMNode(this), s = o == null && !this.props.addEndListener;
    if (!u || s) {
      setTimeout(this.nextCallback, 0);
      return;
    }
    if (this.props.addEndListener) {
      var c = this.props.nodeRef ? [this.nextCallback] : [u, this.nextCallback], f = c[0], m = c[1];
      this.props.addEndListener(f, m);
    }
    o != null && setTimeout(this.nextCallback, o);
  }, r.render = function() {
    var o = this.state.status;
    if (o === En)
      return null;
    var i = this.props, u = i.children;
    i.in, i.mountOnEnter, i.unmountOnExit, i.appear, i.enter, i.exit, i.timeout, i.addEndListener, i.onEnter, i.onEntering, i.onEntered, i.onExit, i.onExiting, i.onExited, i.nodeRef;
    var s = Xi(i, ["children", "in", "mountOnEnter", "unmountOnExit", "appear", "enter", "exit", "timeout", "addEndListener", "onEnter", "onEntering", "onEntered", "onExit", "onExiting", "onExited", "nodeRef"]);
    return (
      // allows for nested Transitions
      /* @__PURE__ */ Ee.createElement(Gt.Provider, {
        value: null
      }, typeof u == "function" ? u(o, s) : Ee.cloneElement(Ee.Children.only(u), s))
    );
  }, n;
}(Ee.Component);
Le.contextType = Gt;
Le.propTypes = process.env.NODE_ENV !== "production" ? {
  /**
   * A React reference to DOM element that need to transition:
   * https://stackoverflow.com/a/51127130/4671932
   *
   *   - When `nodeRef` prop is used, `node` is not passed to callback functions
   *      (e.g. `onEnter`) because user already has direct access to the node.
   *   - When changing `key` prop of `Transition` in a `TransitionGroup` a new
   *     `nodeRef` need to be provided to `Transition` with changed `key` prop
   *     (see
   *     [test/CSSTransition-test.js](https://github.com/reactjs/react-transition-group/blob/13435f897b3ab71f6e19d724f145596f5910581c/test/CSSTransition-test.js#L362-L437)).
   */
  nodeRef: F.shape({
    current: typeof Element > "u" ? F.any : function(e, n, r, t, o, i) {
      var u = e[n];
      return F.instanceOf(u && "ownerDocument" in u ? u.ownerDocument.defaultView.Element : Element)(e, n, r, t, o, i);
    }
  }),
  /**
   * A `function` child can be used instead of a React element. This function is
   * called with the current transition status (`'entering'`, `'entered'`,
   * `'exiting'`, `'exited'`), which can be used to apply context
   * specific props to a component.
   *
   * ```jsx
   * <Transition in={this.state.in} timeout={150}>
   *   {state => (
   *     <MyComponent className={`fade fade-${state}`} />
   *   )}
   * </Transition>
   * ```
   */
  children: F.oneOfType([F.func.isRequired, F.element.isRequired]).isRequired,
  /**
   * Show the component; triggers the enter or exit states
   */
  in: F.bool,
  /**
   * By default the child component is mounted immediately along with
   * the parent `Transition` component. If you want to "lazy mount" the component on the
   * first `in={true}` you can set `mountOnEnter`. After the first enter transition the component will stay
   * mounted, even on "exited", unless you also specify `unmountOnExit`.
   */
  mountOnEnter: F.bool,
  /**
   * By default the child component stays mounted after it reaches the `'exited'` state.
   * Set `unmountOnExit` if you'd prefer to unmount the component after it finishes exiting.
   */
  unmountOnExit: F.bool,
  /**
   * By default the child component does not perform the enter transition when
   * it first mounts, regardless of the value of `in`. If you want this
   * behavior, set both `appear` and `in` to `true`.
   *
   * > **Note**: there are no special appear states like `appearing`/`appeared`, this prop
   * > only adds an additional enter transition. However, in the
   * > `<CSSTransition>` component that first enter transition does result in
   * > additional `.appear-*` classes, that way you can choose to style it
   * > differently.
   */
  appear: F.bool,
  /**
   * Enable or disable enter transitions.
   */
  enter: F.bool,
  /**
   * Enable or disable exit transitions.
   */
  exit: F.bool,
  /**
   * The duration of the transition, in milliseconds.
   * Required unless `addEndListener` is provided.
   *
   * You may specify a single timeout for all transitions:
   *
   * ```jsx
   * timeout={500}
   * ```
   *
   * or individually:
   *
   * ```jsx
   * timeout={{
   *  appear: 500,
   *  enter: 300,
   *  exit: 500,
   * }}
   * ```
   *
   * - `appear` defaults to the value of `enter`
   * - `enter` defaults to `0`
   * - `exit` defaults to `0`
   *
   * @type {number | { enter?: number, exit?: number, appear?: number }}
   */
  timeout: function(n) {
    var r = Qi;
    n.addEndListener || (r = r.isRequired);
    for (var t = arguments.length, o = new Array(t > 1 ? t - 1 : 0), i = 1; i < t; i++)
      o[i - 1] = arguments[i];
    return r.apply(void 0, [n].concat(o));
  },
  /**
   * Add a custom transition end trigger. Called with the transitioning
   * DOM node and a `done` callback. Allows for more fine grained transition end
   * logic. Timeouts are still used as a fallback if provided.
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed.
   *
   * ```jsx
   * addEndListener={(node, done) => {
   *   // use the css transitionend event to mark the finish of a transition
   *   node.addEventListener('transitionend', done, false);
   * }}
   * ```
   */
  addEndListener: F.func,
  /**
   * Callback fired before the "entering" status is applied. An extra parameter
   * `isAppearing` is supplied to indicate if the enter stage is occurring on the initial mount
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed.
   *
   * @type Function(node: HtmlElement, isAppearing: bool) -> void
   */
  onEnter: F.func,
  /**
   * Callback fired after the "entering" status is applied. An extra parameter
   * `isAppearing` is supplied to indicate if the enter stage is occurring on the initial mount
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed.
   *
   * @type Function(node: HtmlElement, isAppearing: bool)
   */
  onEntering: F.func,
  /**
   * Callback fired after the "entered" status is applied. An extra parameter
   * `isAppearing` is supplied to indicate if the enter stage is occurring on the initial mount
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed.
   *
   * @type Function(node: HtmlElement, isAppearing: bool) -> void
   */
  onEntered: F.func,
  /**
   * Callback fired before the "exiting" status is applied.
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed.
   *
   * @type Function(node: HtmlElement) -> void
   */
  onExit: F.func,
  /**
   * Callback fired after the "exiting" status is applied.
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed.
   *
   * @type Function(node: HtmlElement) -> void
   */
  onExiting: F.func,
  /**
   * Callback fired after the "exited" status is applied.
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed
   *
   * @type Function(node: HtmlElement) -> void
   */
  onExited: F.func
} : {};
function nn() {
}
Le.defaultProps = {
  in: !1,
  mountOnEnter: !1,
  unmountOnExit: !1,
  appear: !1,
  enter: !0,
  exit: !0,
  onEnter: nn,
  onEntering: nn,
  onEntered: nn,
  onExit: nn,
  onExiting: nn,
  onExited: nn
};
Le.UNMOUNTED = En;
Le.EXITED = ze;
Le.ENTERING = Ie;
Le.ENTERED = Je;
Le.EXITING = Tn;
const na = Le;
function lt(e, n) {
  const r = Ge(e, n) || "", t = r.indexOf("ms") === -1 ? 1e3 : 1;
  return parseFloat(r) * t;
}
function ra(e, n) {
  const r = lt(e, "transitionDuration"), t = lt(e, "transitionDelay"), o = qt(e, (i) => {
    i.target === e && (o(), n(i));
  }, r + t);
}
function ta(e) {
  e.offsetHeight;
}
function oa(e) {
  return e && "setState" in e ? rn.findDOMNode(e) : e ?? null;
}
const ia = /* @__PURE__ */ Ee.forwardRef(({
  onEnter: e,
  onEntering: n,
  onEntered: r,
  onExit: t,
  onExiting: o,
  onExited: i,
  addEndListener: u,
  children: s,
  childRef: c,
  ...f
}, m) => {
  const v = he(null), h = Dn(v, c), C = (G) => {
    h(oa(G));
  }, S = (G) => (se) => {
    G && v.current && G(v.current, se);
  }, T = Ce(S(e), [e]), O = Ce(S(n), [n]), A = Ce(S(r), [r]), _ = Ce(S(t), [t]), B = Ce(S(o), [o]), q = Ce(S(i), [i]), E = Ce(S(u), [u]);
  return /* @__PURE__ */ P.jsx(na, {
    ref: m,
    ...f,
    onEnter: T,
    onEntered: A,
    onEntering: O,
    onExit: _,
    onExited: q,
    onExiting: B,
    addEndListener: E,
    nodeRef: v,
    children: typeof s == "function" ? (G, se) => (
      // TODO: Types for RTG missing innerProps, so need to cast.
      s(G, {
        ...se,
        ref: C
      })
    ) : /* @__PURE__ */ Ee.cloneElement(s, {
      ref: C
    })
  });
}), aa = ia, sa = {
  [Ie]: "show",
  [Je]: "show"
}, Kt = /* @__PURE__ */ U.forwardRef(({
  className: e,
  children: n,
  transitionClasses: r = {},
  onEnter: t,
  ...o
}, i) => {
  const u = {
    in: !1,
    timeout: 300,
    mountOnEnter: !1,
    unmountOnExit: !1,
    appear: !1,
    ...o
  }, s = Ce((c, f) => {
    ta(c), t == null || t(c, f);
  }, [t]);
  return /* @__PURE__ */ P.jsx(aa, {
    ref: i,
    addEndListener: ra,
    ...u,
    onEnter: s,
    childRef: n.ref,
    children: (c, f) => /* @__PURE__ */ U.cloneElement(n, {
      ...f,
      className: re("fade", e, n.props.className, sa[c], r[c])
    })
  });
});
Kt.displayName = "Fade";
const Or = Kt, Xt = /* @__PURE__ */ U.forwardRef(({
  className: e,
  bsPrefix: n,
  as: r = "div",
  ...t
}, o) => (n = de(n, "modal-body"), /* @__PURE__ */ P.jsx(r, {
  ref: o,
  className: re(e, n),
  ...t
})));
Xt.displayName = "ModalBody";
const ua = Xt, la = /* @__PURE__ */ U.createContext({
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  onHide() {
  }
}), Zt = la, Qt = /* @__PURE__ */ U.forwardRef(({
  bsPrefix: e,
  className: n,
  contentClassName: r,
  centered: t,
  size: o,
  fullscreen: i,
  children: u,
  scrollable: s,
  ...c
}, f) => {
  e = de(e, "modal");
  const m = `${e}-dialog`, v = typeof i == "string" ? `${e}-fullscreen-${i}` : `${e}-fullscreen`;
  return /* @__PURE__ */ P.jsx("div", {
    ...c,
    ref: f,
    className: re(m, n, o && `${e}-${o}`, t && `${m}-centered`, s && `${m}-scrollable`, i && v),
    children: /* @__PURE__ */ P.jsx("div", {
      className: re(`${e}-content`, r),
      children: u
    })
  });
});
Qt.displayName = "ModalDialog";
const eo = Qt, no = /* @__PURE__ */ U.forwardRef(({
  className: e,
  bsPrefix: n,
  as: r = "div",
  ...t
}, o) => (n = de(n, "modal-footer"), /* @__PURE__ */ P.jsx(r, {
  ref: o,
  className: re(e, n),
  ...t
})));
no.displayName = "ModalFooter";
const ca = no, fa = {
  /** An accessible label indicating the relevant information about the Close Button. */
  "aria-label": F.string,
  /** A callback fired after the Close Button is clicked. */
  onClick: F.func,
  /**
   * Render different color variant for the button.
   *
   * Omitting this will render the default dark color.
   */
  variant: F.oneOf(["white"])
}, kr = /* @__PURE__ */ U.forwardRef(({
  className: e,
  variant: n,
  "aria-label": r = "Close",
  ...t
}, o) => /* @__PURE__ */ P.jsx("button", {
  ref: o,
  type: "button",
  className: re("btn-close", n && `btn-close-${n}`, e),
  "aria-label": r,
  ...t
}));
kr.displayName = "CloseButton";
kr.propTypes = fa;
const ro = kr, da = /* @__PURE__ */ U.forwardRef(({
  closeLabel: e = "Close",
  closeVariant: n,
  closeButton: r = !1,
  onHide: t,
  children: o,
  ...i
}, u) => {
  const s = Ke(Zt), c = Ae(() => {
    s == null || s.onHide(), t == null || t();
  });
  return /* @__PURE__ */ P.jsxs("div", {
    ref: u,
    ...i,
    children: [o, r && /* @__PURE__ */ P.jsx(ro, {
      "aria-label": e,
      variant: n,
      onClick: c
    })]
  });
}), ma = da, to = /* @__PURE__ */ U.forwardRef(({
  bsPrefix: e,
  className: n,
  closeLabel: r = "Close",
  closeButton: t = !1,
  ...o
}, i) => (e = de(e, "modal-header"), /* @__PURE__ */ P.jsx(ma, {
  ref: i,
  ...o,
  className: re(n, e),
  closeLabel: r,
  closeButton: t
})));
to.displayName = "ModalHeader";
const pa = to, va = jr("h4"), oo = /* @__PURE__ */ U.forwardRef(({
  className: e,
  bsPrefix: n,
  as: r = va,
  ...t
}, o) => (n = de(n, "modal-title"), /* @__PURE__ */ P.jsx(r, {
  ref: o,
  className: re(e, n),
  ...t
})));
oo.displayName = "ModalTitle";
const ha = oo;
function ba(e) {
  return /* @__PURE__ */ P.jsx(Or, {
    ...e,
    timeout: null
  });
}
function ya(e) {
  return /* @__PURE__ */ P.jsx(Or, {
    ...e,
    timeout: null
  });
}
const io = /* @__PURE__ */ U.forwardRef(({
  bsPrefix: e,
  className: n,
  style: r,
  dialogClassName: t,
  contentClassName: o,
  children: i,
  dialogAs: u = eo,
  "aria-labelledby": s,
  "aria-describedby": c,
  "aria-label": f,
  /* BaseModal props */
  show: m = !1,
  animation: v = !0,
  backdrop: h = !0,
  keyboard: C = !0,
  onEscapeKeyDown: S,
  onShow: T,
  onHide: O,
  container: A,
  autoFocus: _ = !0,
  enforceFocus: B = !0,
  restoreFocus: q = !0,
  restoreFocusOptions: E,
  onEntered: G,
  onExit: se,
  onExiting: ye,
  onEnter: xe,
  onEntering: we,
  onExited: be,
  backdropClassName: ge,
  manager: Z,
  ...te
}, le) => {
  const [ne, pe] = fe({}), [oe, Re] = fe(!1), y = he(!1), w = he(!1), M = he(null), [V, $] = pi(), J = Dn(le, $), I = Ae(O), W = Wo();
  e = de(e, "modal");
  const Y = tn(() => ({
    onHide: I
  }), [I]);
  function H() {
    return Z || Ki({
      isRTL: W
    });
  }
  function L(X) {
    if (!on)
      return;
    const Oe = H().getScrollbarWidth() > 0, _e = X.scrollHeight > kn(X).documentElement.clientHeight;
    pe({
      paddingRight: Oe && !_e ? nt() : void 0,
      paddingLeft: !Oe && _e ? nt() : void 0
    });
  }
  const ue = Ae(() => {
    V && L(V.dialog);
  });
  Sr(() => {
    br(window, "resize", ue), M.current == null || M.current();
  });
  const N = () => {
    y.current = !0;
  }, ve = (X) => {
    y.current && V && X.target === V.dialog && (w.current = !0), y.current = !1;
  }, Ne = () => {
    Re(!0), M.current = qt(V.dialog, () => {
      Re(!1);
    });
  }, K = (X) => {
    X.target === X.currentTarget && Ne();
  }, je = (X) => {
    if (h === "static") {
      K(X);
      return;
    }
    if (w.current || X.target !== X.currentTarget) {
      w.current = !1;
      return;
    }
    O == null || O();
  }, ke = (X) => {
    C ? S == null || S(X) : (X.preventDefault(), h === "static" && Ne());
  }, We = (X, Oe) => {
    X && L(X), xe == null || xe(X, Oe);
  }, De = (X) => {
    M.current == null || M.current(), se == null || se(X);
  }, sn = (X, Oe) => {
    we == null || we(X, Oe), Yt(window, "resize", ue);
  }, un = (X) => {
    X && (X.style.display = ""), be == null || be(X), br(window, "resize", ue);
  }, Pe = Ce((X) => /* @__PURE__ */ P.jsx("div", {
    ...X,
    className: re(`${e}-backdrop`, ge, !v && "show")
  }), [v, ge, e]), Xe = {
    ...r,
    ...ne
  };
  Xe.display = "block";
  const ln = (X) => /* @__PURE__ */ P.jsx("div", {
    role: "dialog",
    ...X,
    style: Xe,
    className: re(n, e, oe && `${e}-static`, !v && "show"),
    onClick: h ? je : void 0,
    onMouseUp: ve,
    "aria-label": f,
    "aria-labelledby": s,
    "aria-describedby": c,
    children: /* @__PURE__ */ P.jsx(u, {
      ...te,
      onMouseDown: N,
      className: t,
      contentClassName: o,
      children: i
    })
  });
  return /* @__PURE__ */ P.jsx(Zt.Provider, {
    value: Y,
    children: /* @__PURE__ */ P.jsx(Yi, {
      show: m,
      ref: J,
      backdrop: h,
      container: A,
      keyboard: !0,
      autoFocus: _,
      enforceFocus: B,
      restoreFocus: q,
      restoreFocusOptions: E,
      onEscapeKeyDown: ke,
      onShow: T,
      onHide: O,
      onEnter: We,
      onEntering: sn,
      onEntered: G,
      onExit: De,
      onExiting: ye,
      onExited: un,
      manager: H(),
      transition: v ? ba : void 0,
      backdropTransition: v ? ya : void 0,
      renderBackdrop: Pe,
      renderDialog: ln
    })
  });
});
io.displayName = "Modal";
const gn = Object.assign(io, {
  Body: ua,
  Header: pa,
  Title: ha,
  Footer: ca,
  Dialog: eo,
  TRANSITION_DURATION: 300,
  BACKDROP_TRANSITION_DURATION: 150
}), ao = /* @__PURE__ */ U.forwardRef(({
  bsPrefix: e,
  className: n,
  // Need to define the default "as" during prop destructuring to be compatible with styled-components github.com/react-bootstrap/react-bootstrap/issues/3595
  as: r = "div",
  ...t
}, o) => {
  const i = de(e, "row"), u = jt(), s = _t(), c = `${i}-cols`, f = [];
  return u.forEach((m) => {
    const v = t[m];
    delete t[m];
    let h;
    v != null && typeof v == "object" ? {
      cols: h
    } = v : h = v;
    const C = m !== s ? `-${m}` : "";
    h != null && f.push(`${c}${C}-${h}`);
  }), /* @__PURE__ */ P.jsx(r, {
    ref: o,
    ...t,
    className: re(n, i, ...f)
  });
});
ao.displayName = "Row";
const an = ao;
function ga({
  as: e,
  bsPrefix: n,
  className: r,
  ...t
}) {
  n = de(n, "col");
  const o = jt(), i = _t(), u = [], s = [];
  return o.forEach((c) => {
    const f = t[c];
    delete t[c];
    let m, v, h;
    typeof f == "object" && f != null ? {
      span: m,
      offset: v,
      order: h
    } = f : m = f;
    const C = c !== i ? `-${c}` : "";
    m && u.push(m === !0 ? `${n}${C}` : `${n}${C}-${m}`), h != null && s.push(`order${C}-${h}`), v != null && s.push(`offset${C}-${v}`);
  }), [{
    ...t,
    className: re(r, ...u, ...s)
  }, {
    as: e,
    bsPrefix: n,
    spans: u
  }];
}
const so = /* @__PURE__ */ U.forwardRef(
  // Need to define the default "as" during prop destructuring to be compatible with styled-components github.com/react-bootstrap/react-bootstrap/issues/3595
  (e, n) => {
    const [{
      className: r,
      ...t
    }, {
      as: o = "div",
      bsPrefix: i,
      spans: u
    }] = ga(e);
    return /* @__PURE__ */ P.jsx(o, {
      ...t,
      ref: n,
      className: re(r, !u.length && i)
    });
  }
);
so.displayName = "Col";
const Ve = so, uo = ({ children: e, title: n = "Dialog", oklabel: r = "Ok", cancellabel: t = "Zruit", onOk: o, onCancel: i }) => {
  const u = () => {
    o && o();
  }, s = () => {
    i && i();
  };
  return (
    // <div className="modal show" style={{ display: 'block', position: 'initial' }} >
    /* @__PURE__ */ d.jsxDEV(gn, { size: "xl", show: "true", onHide: s, children: [
      /* @__PURE__ */ d.jsxDEV(gn.Header, { closeButton: !0, children: /* @__PURE__ */ d.jsxDEV(gn.Title, { children: n }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Dialog.jsx",
        lineNumber: 28,
        columnNumber: 17
      }, void 0) }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Dialog.jsx",
        lineNumber: 27,
        columnNumber: 13
      }, void 0),
      /* @__PURE__ */ d.jsxDEV(gn.Body, { children: e }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Dialog.jsx",
        lineNumber: 30,
        columnNumber: 13
      }, void 0),
      /* @__PURE__ */ d.jsxDEV(gn.Footer, { children: /* @__PURE__ */ d.jsxDEV(an, { children: /* @__PURE__ */ d.jsxDEV(Ve, { children: /* @__PURE__ */ d.jsxDEV("div", { className: "d-flex justify-content-end", children: [
        /* @__PURE__ */ d.jsxDEV(et, { variant: "outline-success", onClick: u, children: r }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Dialog.jsx",
          lineNumber: 37,
          columnNumber: 29
        }, void 0),
        " ",
        /* @__PURE__ */ d.jsxDEV(et, { variant: "outline-danger", onClick: s, children: t }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Dialog.jsx",
          lineNumber: 38,
          columnNumber: 29
        }, void 0)
      ] }, void 0, !0, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Dialog.jsx",
        lineNumber: 36,
        columnNumber: 25
      }, void 0) }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Dialog.jsx",
        lineNumber: 35,
        columnNumber: 21
      }, void 0) }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Dialog.jsx",
        lineNumber: 34,
        columnNumber: 17
      }, void 0) }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Dialog.jsx",
        lineNumber: 33,
        columnNumber: 13
      }, void 0)
    ] }, void 0, !0, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Dialog.jsx",
      lineNumber: 26,
      columnNumber: 9
    }, void 0)
  );
};
uo.propTypes = {
  children: F.oneOfType([
    F.arrayOf(F.node),
    F.node
  ]),
  title: F.string,
  oklabel: F.string,
  cancellabel: F.string,
  onOk: F.func.isRequired,
  onCancel: F.func.isRequired
};
var gr = { exports: {} }, ur = {};
/**
 * @license React
 * use-sync-external-store-with-selector.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ct;
function Ea() {
  return ct || (ct = 1, process.env.NODE_ENV !== "production" && function() {
    typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ < "u" && typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStart == "function" && __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStart(new Error());
    var e = Ee;
    function n(f, m) {
      return f === m && (f !== 0 || 1 / f === 1 / m) || f !== f && m !== m;
    }
    var r = typeof Object.is == "function" ? Object.is : n, t = e.useSyncExternalStore, o = e.useRef, i = e.useEffect, u = e.useMemo, s = e.useDebugValue;
    function c(f, m, v, h, C) {
      var S = o(null), T;
      S.current === null ? (T = {
        hasValue: !1,
        value: null
      }, S.current = T) : T = S.current;
      var O = u(function() {
        var q = !1, E, G, se = function(be) {
          if (!q) {
            q = !0, E = be;
            var ge = h(be);
            if (C !== void 0 && T.hasValue) {
              var Z = T.value;
              if (C(Z, ge))
                return G = Z, Z;
            }
            return G = ge, ge;
          }
          var te = E, le = G;
          if (r(te, be))
            return le;
          var ne = h(be);
          return C !== void 0 && C(le, ne) ? le : (E = be, G = ne, ne);
        }, ye = v === void 0 ? null : v, xe = function() {
          return se(m());
        }, we = ye === null ? void 0 : function() {
          return se(ye());
        };
        return [xe, we];
      }, [m, v, h, C]), A = O[0], _ = O[1], B = t(f, A, _);
      return i(function() {
        T.hasValue = !0, T.value = B;
      }, [B]), s(B), B;
    }
    ur.useSyncExternalStoreWithSelector = c, typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ < "u" && typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStop == "function" && __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStop(new Error());
  }()), ur;
}
var lr = {};
/**
 * @license React
 * use-sync-external-store-with-selector.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ft;
function xa() {
  if (ft)
    return lr;
  ft = 1;
  var e = Ee;
  function n(c, f) {
    return c === f && (c !== 0 || 1 / c === 1 / f) || c !== c && f !== f;
  }
  var r = typeof Object.is == "function" ? Object.is : n, t = e.useSyncExternalStore, o = e.useRef, i = e.useEffect, u = e.useMemo, s = e.useDebugValue;
  return lr.useSyncExternalStoreWithSelector = function(c, f, m, v, h) {
    var C = o(null);
    if (C.current === null) {
      var S = { hasValue: !1, value: null };
      C.current = S;
    } else
      S = C.current;
    C = u(function() {
      function O(E) {
        if (!A) {
          if (A = !0, _ = E, E = v(E), h !== void 0 && S.hasValue) {
            var G = S.value;
            if (h(G, E))
              return B = G;
          }
          return B = E;
        }
        if (G = B, r(_, E))
          return G;
        var se = v(E);
        return h !== void 0 && h(G, se) ? G : (_ = E, B = se);
      }
      var A = !1, _, B, q = m === void 0 ? null : m;
      return [function() {
        return O(f());
      }, q === null ? void 0 : function() {
        return O(q());
      }];
    }, [f, m, v, h]);
    var T = t(c, C[0], C[1]);
    return i(function() {
      S.hasValue = !0, S.value = T;
    }, [T]), s(T), T;
  }, lr;
}
process.env.NODE_ENV === "production" ? gr.exports = xa() : gr.exports = Ea();
var Na = gr.exports, Fe = (
  // prettier-ignore
  // @ts-ignore
  "default" in U ? U.default : U
), dt = Symbol.for("react-redux-context"), mt = typeof globalThis < "u" ? globalThis : (
  /* fall back to a per-module scope (pre-8.1 behaviour) if `globalThis` is not available */
  {}
);
function wa() {
  if (!Fe.createContext)
    return {};
  const e = mt[dt] ?? (mt[dt] = /* @__PURE__ */ new Map());
  let n = e.get(Fe.createContext);
  return n || (n = Fe.createContext(
    null
  ), process.env.NODE_ENV !== "production" && (n.displayName = "ReactRedux"), e.set(Fe.createContext, n)), n;
}
var Be = /* @__PURE__ */ wa(), Ca = () => {
  throw new Error("uSES not initialized!");
};
function Dr(e = Be) {
  return function() {
    const r = Fe.useContext(e);
    if (process.env.NODE_ENV !== "production" && !r)
      throw new Error(
        "could not find react-redux context value; please ensure the component is wrapped in a <Provider>"
      );
    return r;
  };
}
var lo = /* @__PURE__ */ Dr(), co = Ca, Ra = (e) => {
  co = e;
}, ja = (e, n) => e === n;
function _a(e = Be) {
  const n = e === Be ? lo : Dr(e);
  return function(t, o = {}) {
    const { equalityFn: i = ja, devModeChecks: u = {} } = typeof o == "function" ? { equalityFn: o } : o;
    if (process.env.NODE_ENV !== "production") {
      if (!t)
        throw new Error("You must pass a selector to useSelector");
      if (typeof t != "function")
        throw new Error("You must pass a function as a selector to useSelector");
      if (typeof i != "function")
        throw new Error(
          "You must pass a function as an equality function to useSelector"
        );
    }
    const {
      store: s,
      subscription: c,
      getServerState: f,
      stabilityCheck: m,
      identityFunctionCheck: v
    } = n(), h = Fe.useRef(!0), C = Fe.useCallback(
      {
        [t.name](T) {
          const O = t(T);
          if (process.env.NODE_ENV !== "production") {
            const {
              identityFunctionCheck: A,
              stabilityCheck: _
            } = {
              stabilityCheck: m,
              identityFunctionCheck: v,
              ...u
            };
            if (_ === "always" || _ === "once" && h.current) {
              const B = t(T);
              if (!i(O, B)) {
                let q;
                try {
                  throw new Error();
                } catch (E) {
                  ({ stack: q } = E);
                }
                console.warn(
                  "Selector " + (t.name || "unknown") + ` returned a different result when called with the same parameters. This can lead to unnecessary rerenders.
Selectors that return a new reference (such as an object or an array) should be memoized: https://redux.js.org/usage/deriving-data-selectors#optimizing-selectors-with-memoization`,
                  {
                    state: T,
                    selected: O,
                    selected2: B,
                    stack: q
                  }
                );
              }
            }
            if ((A === "always" || A === "once" && h.current) && O === T) {
              let B;
              try {
                throw new Error();
              } catch (q) {
                ({ stack: B } = q);
              }
              console.warn(
                "Selector " + (t.name || "unknown") + ` returned the root state when called. This can lead to unnecessary rerenders.
Selectors that return the entire state are almost certainly a mistake, as they will cause a rerender whenever *anything* in state changes.`,
                { stack: B }
              );
            }
            h.current && (h.current = !1);
          }
          return O;
        }
      }[t.name],
      [t, m, u.stabilityCheck]
    ), S = co(
      c.addNestedSub,
      s.getState,
      f || s.getState,
      C,
      i
    );
    return Fe.useDebugValue(S), S;
  };
}
var An = /* @__PURE__ */ _a();
function Sa(e) {
  e();
}
function Ta() {
  let e = null, n = null;
  return {
    clear() {
      e = null, n = null;
    },
    notify() {
      Sa(() => {
        let r = e;
        for (; r; )
          r.callback(), r = r.next;
      });
    },
    get() {
      let r = [], t = e;
      for (; t; )
        r.push(t), t = t.next;
      return r;
    },
    subscribe(r) {
      let t = !0, o = n = {
        callback: r,
        next: null,
        prev: n
      };
      return o.prev ? o.prev.next = o : e = o, function() {
        !t || e === null || (t = !1, o.next ? o.next.prev = o.prev : n = o.prev, o.prev ? o.prev.next = o.next : e = o.next);
      };
    }
  };
}
var pt = {
  notify() {
  },
  get: () => []
};
function Fa(e, n) {
  let r, t = pt, o = 0, i = !1;
  function u(T) {
    m();
    const O = t.subscribe(T);
    let A = !1;
    return () => {
      A || (A = !0, O(), v());
    };
  }
  function s() {
    t.notify();
  }
  function c() {
    S.onStateChange && S.onStateChange();
  }
  function f() {
    return i;
  }
  function m() {
    o++, r || (r = n ? n.addNestedSub(c) : e.subscribe(c), t = Ta());
  }
  function v() {
    o--, r && o === 0 && (r(), r = void 0, t.clear(), t = pt);
  }
  function h() {
    i || (i = !0, m());
  }
  function C() {
    i && (i = !1, v());
  }
  const S = {
    addNestedSub: u,
    notifyNestedSubs: s,
    handleChangeWrapper: c,
    isSubscribed: f,
    trySubscribe: h,
    tryUnsubscribe: C,
    getListeners: () => t
  };
  return S;
}
var Oa = typeof window < "u" && typeof window.document < "u" && typeof window.document.createElement < "u", ka = Oa ? Fe.useLayoutEffect : Fe.useEffect;
function Da({
  store: e,
  context: n,
  children: r,
  serverState: t,
  stabilityCheck: o = "once",
  identityFunctionCheck: i = "once"
}) {
  const u = Fe.useMemo(() => {
    const f = Fa(e);
    return {
      store: e,
      subscription: f,
      getServerState: t ? () => t : void 0,
      stabilityCheck: o,
      identityFunctionCheck: i
    };
  }, [e, t, o, i]), s = Fe.useMemo(() => e.getState(), [e]);
  ka(() => {
    const { subscription: f } = u;
    return f.onStateChange = f.notifyNestedSubs, f.trySubscribe(), s !== e.getState() && f.notifyNestedSubs(), () => {
      f.tryUnsubscribe(), f.onStateChange = void 0;
    };
  }, [u, s]);
  const c = n || Be;
  return /* @__PURE__ */ Fe.createElement(c.Provider, { value: u }, r);
}
var Aa = Da;
function fo(e = Be) {
  const n = (
    // @ts-ignore
    e === Be ? lo : (
      // @ts-ignore
      Dr(e)
    )
  );
  return function() {
    const { store: t } = n();
    return t;
  };
}
var Va = /* @__PURE__ */ fo();
function Pa(e = Be) {
  const n = (
    // @ts-ignore
    e === Be ? Va : fo(e)
  );
  return function() {
    return n().dispatch;
  };
}
var $e = /* @__PURE__ */ Pa();
Ra(Na.useSyncExternalStoreWithSelector);
const mo = (e, n) => {
  const r = n.payload, t = r.id || crypto.randomUUID();
  return r.id || (r.id = t), e[t] = r, e;
}, po = (e, n) => {
  const r = n.payload;
  return delete e[r.id], e;
}, Ma = (e, n) => {
  const r = n.payload;
  return e[r.id] = r, e;
}, Ia = (e, n) => {
  const r = n.payload, t = e[r.id];
  return e[r.id] = { ...t, ...r }, e;
}, $a = (e, n) => {
  const { item: r, vectorname: t } = n.payload, o = e[r.id], i = t in o ? o[t] : [], u = {};
  for (let s of i)
    u[s.id] = s;
  console.log("UpdateSubVector", r), console.log("UpdateSubVector", r), console.log("UpdateSubVector", r[t]);
  for (let s of r[t]) {
    s != null && s.__typename && (e[s.id] = { ...e[s.id], ...s });
    const c = u[s.id] || {};
    u[s.id] = { ...c, ...s };
  }
  return o[t] = Object.values(u), e;
}, La = (e, n) => {
  const { item: r, scalarname: t } = n.payload, o = e[r.id], i = r[t];
  return o[t] = { ...o[t], ...i }, i != null && i.__typename && (e[i.id] = { ...e[i.id], ...i }), e;
}, Ba = mo, Wa = Ia, Ua = Ma, Ya = po, qa = La, Ha = $a, vo = wt({
  name: "items",
  initialState: {},
  reducers: {
    item_add: Ba,
    item_update: Wa,
    item_replace: Ua,
    item_delete: Ya,
    item_updateAttributeScalar: qa,
    item_updateAttributeVector: Ha
  }
}), za = vo.reducer, Er = vo.actions, xr = 2 ** 31 - 1;
function ho(e, n, r) {
  const t = r - Date.now();
  e.current = t <= xr ? setTimeout(n, t) : setTimeout(() => ho(e, n, r), xr);
}
function Ja() {
  const e = Ht(), n = he();
  return Sr(() => clearTimeout(n.current)), tn(() => {
    const r = () => clearTimeout(n.current);
    function t(o, i = 0) {
      e() && (r(), i <= xr ? n.current = setTimeout(o, i) : ho(n, o, Date.now() + i));
    }
    return {
      set: t,
      clear: r,
      handleRef: n
    };
  }, []);
}
const Ga = {
  [Ie]: "showing",
  [Tn]: "showing show"
}, bo = /* @__PURE__ */ U.forwardRef((e, n) => /* @__PURE__ */ P.jsx(Or, {
  ...e,
  ref: n,
  transitionClasses: Ga
}));
bo.displayName = "ToastFade";
const Ka = bo, Xa = /* @__PURE__ */ U.createContext({
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  onClose() {
  }
}), yo = Xa, go = /* @__PURE__ */ U.forwardRef(({
  bsPrefix: e,
  closeLabel: n = "Close",
  closeVariant: r,
  closeButton: t = !0,
  className: o,
  children: i,
  ...u
}, s) => {
  e = de(e, "toast-header");
  const c = Ke(yo), f = Ae((m) => {
    c == null || c.onClose == null || c.onClose(m);
  });
  return /* @__PURE__ */ P.jsxs("div", {
    ref: s,
    ...u,
    className: re(e, o),
    children: [i, t && /* @__PURE__ */ P.jsx(ro, {
      "aria-label": n,
      variant: r,
      onClick: f,
      "data-dismiss": "toast"
    })]
  });
});
go.displayName = "ToastHeader";
const Za = go, Eo = /* @__PURE__ */ U.forwardRef(({
  className: e,
  bsPrefix: n,
  as: r = "div",
  ...t
}, o) => (n = de(n, "toast-body"), /* @__PURE__ */ P.jsx(r, {
  ref: o,
  className: re(e, n),
  ...t
})));
Eo.displayName = "ToastBody";
const Qa = Eo, xo = /* @__PURE__ */ U.forwardRef(({
  bsPrefix: e,
  className: n,
  transition: r = Ka,
  show: t = !0,
  animation: o = !0,
  delay: i = 5e3,
  autohide: u = !1,
  onClose: s,
  onEntered: c,
  onExit: f,
  onExiting: m,
  onEnter: v,
  onEntering: h,
  onExited: C,
  bg: S,
  ...T
}, O) => {
  e = de(e, "toast");
  const A = he(i), _ = he(s);
  Se(() => {
    A.current = i, _.current = s;
  }, [i, s]);
  const B = Ja(), q = !!(u && t), E = Ce(() => {
    q && (_.current == null || _.current());
  }, [q]);
  Se(() => {
    B.set(E, A.current);
  }, [B, E]);
  const G = tn(() => ({
    onClose: s
  }), [s]), se = !!(r && o), ye = /* @__PURE__ */ P.jsx("div", {
    ...T,
    ref: O,
    className: re(e, n, S && `bg-${S}`, !se && (t ? "show" : "hide")),
    role: "alert",
    "aria-live": "assertive",
    "aria-atomic": "true"
  });
  return /* @__PURE__ */ P.jsx(yo.Provider, {
    value: G,
    children: se && r ? /* @__PURE__ */ P.jsx(r, {
      in: t,
      onEnter: v,
      onEntering: h,
      onEntered: c,
      onExit: f,
      onExiting: m,
      onExited: C,
      unmountOnExit: !0,
      children: ye
    }) : ye
  });
});
xo.displayName = "Toast";
const cr = Object.assign(xo, {
  Body: Qa,
  Header: Za
}), es = {
  "top-start": "top-0 start-0",
  "top-center": "top-0 start-50 translate-middle-x",
  "top-end": "top-0 end-0",
  "middle-start": "top-50 start-0 translate-middle-y",
  "middle-center": "top-50 start-50 translate-middle",
  "middle-end": "top-50 end-0 translate-middle-y",
  "bottom-start": "bottom-0 start-0",
  "bottom-center": "bottom-0 start-50 translate-middle-x",
  "bottom-end": "bottom-0 end-0"
}, No = /* @__PURE__ */ U.forwardRef(({
  bsPrefix: e,
  position: n,
  containerPosition: r,
  className: t,
  // Need to define the default "as" during prop destructuring to be compatible with styled-components github.com/react-bootstrap/react-bootstrap/issues/3595
  as: o = "div",
  ...i
}, u) => (e = de(e, "toast-container"), /* @__PURE__ */ P.jsx(o, {
  ref: u,
  ...i,
  className: re(e, n && es[n], r && `position-${r}`, t)
})));
No.displayName = "ToastContainer";
const ns = No, wo = wt({
  name: "msgs",
  initialState: {},
  reducers: {
    msg_add: mo,
    msg_delete: po
  }
}), rs = wo.reducer, Fn = wo.actions, ts = ({ msg: e }) => {
  const n = $e(), r = () => {
    n(Fn.msg_delete(e));
  };
  return e != null && e.variant && "" + e.variant, /* @__PURE__ */ d.jsxDEV(cr, { bg: e.variant, onClose: r, children: [
    /* @__PURE__ */ d.jsxDEV(cr.Header, {}, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Store/msgs.jsx",
      lineNumber: 55,
      columnNumber: 13
    }, void 0),
    /* @__PURE__ */ d.jsxDEV(cr.Body, { children: e == null ? void 0 : e.title }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Store/msgs.jsx",
      lineNumber: 62,
      columnNumber: 13
    }, void 0)
  ] }, void 0, !0, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Store/msgs.jsx",
    lineNumber: 54,
    columnNumber: 9
  }, void 0);
}, os = () => {
  const e = An((n) => n.msgs);
  return /* @__PURE__ */ d.jsxDEV(ns, { position: "bottom-end", children: Object.values(e).map(
    (n) => /* @__PURE__ */ d.jsxDEV(ts, { msg: n }, n.id, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Store/msgs.jsx",
      lineNumber: 85,
      columnNumber: 24
    }, void 0)
  ) }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Store/msgs.jsx",
    lineNumber: 83,
    columnNumber: 9
  }, void 0);
}, Co = () => crypto.randomUUID(), vt = ({ title: e, delay: n = 5e3, variant: r = "success", detail: t = [] }) => (o, i) => {
  const u = { id: Co(), variant: r, title: e, detail: t };
  return setTimeout(
    () => o(Fn.msg_delete(u)),
    n
  ), o(Fn.msg_add(u));
}, fr = ({ title: e, variant: n = "danger", detail: r = [] }) => (t, o) => {
  const i = { id: Co(), variant: n, title: e, detail: r };
  return t(Fn.msg_add(i));
}, Ar = (e) => (n) => [(o) => {
  const i = o == null ? void 0 : o.errors;
  i && n(fr({ title: e.error, variant: "danger", detail: i }));
  const u = o == null ? void 0 : o.data;
  if (u) {
    const s = u == null ? void 0 : u.result;
    s && (s != null && s.msg ? (s == null ? void 0 : s.msg) !== "ok" ? n(fr({ title: e.error, variant: "danger" })) : n(vt({ title: e.success, variant: "success" })) : n(vt({ title: e.success, variant: "success" })));
  }
  return o;
}, (o) => (console.log("CreateAsyncQueryValidator.onReject", o), n(fr({ title: e.error, variant: "danger", detail: ["" + o] })), o)], Nu = (e) => (n) => {
  var t, o, i;
  if (console.log("CheckGQLError call"), n == null ? void 0 : n.errors)
    ((e == null ? void 0 : e.errors) || (e == null ? void 0 : e.error) || e.fail)(n);
  else {
    const u = (o = (t = n == null ? void 0 : n.data) == null ? void 0 : t.result) == null ? void 0 : o.msg;
    if (u) {
      const s = e[u] || e.fail;
      s ? s(n) : console.warn("no proper reaction found", n);
    } else if ((i = n == null ? void 0 : n.data) == null ? void 0 : i.result) {
      const c = e.ok;
      c ? c(n) : console.error("ok reaction missing");
    } else {
      const c = (e == null ? void 0 : e.errors) || (e == null ? void 0 : e.error) || (e == null ? void 0 : e.fail);
      c && c(`Data nenalezena. ${JSON.stringify(n)}  Server nenael poloku v databzi. Poloka neexistuje.`), console.warn("no proper reaction found", n);
    }
  }
  return n;
}, is = Ar({ error: "Nco se nepovedlo", success: "Zmna uloena" }), wu = ({ item: e, label: n, children: r, attributeName: t, asyncUpdater: o }) => {
  const i = $e(), u = e[t], [s, c] = is(i), f = (m) => {
    const v = m.target.value, h = { ...e };
    h[t] = v;
    const C = o({ ...h });
    i(C).then(s, c);
  };
  return (
    // <EditableText id={item.id} value={attributeValue} onChange={onChange_} />
    /* @__PURE__ */ d.jsxDEV("div", { className: "form-floating", children: [
      /* @__PURE__ */ d.jsxDEV("select", { className: "form-select", id: "select-" + e.id, value: u, onChange: f, "aria-label": "", children: r }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/EditableAttributeSelect.jsx",
        lineNumber: 35,
        columnNumber: 13
      }, void 0),
      /* @__PURE__ */ d.jsxDEV("label", { htmlFor: "select-" + e.id, children: n }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/EditableAttributeSelect.jsx",
        lineNumber: 38,
        columnNumber: 13
      }, void 0)
    ] }, void 0, !0, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/EditableAttributeSelect.jsx",
      lineNumber: 34,
      columnNumber: 9
    }, void 0)
  );
}, Ro = ({ id: e, value: n, onChange: r, placeholder: t, type: o = "text" }) => {
  const [i, u] = fe(n), s = tn(
    () => ei(),
    [e]
  ), c = (
    //useCallback(
    (m) => {
      const v = m.target.value;
      u(v), r && s(() => r(v));
    }
  ), f = (
    //useCallback(
    (m) => {
      const v = m.target.value;
      v !== i && (c(m), r(v));
    }
  );
  return /* @__PURE__ */ d.jsxDEV("input", { type: o, id: e, className: "form-control", placeholder: t, value: i, onChange: c, onBlur: f }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/TextInput.jsx",
    lineNumber: 48,
    columnNumber: 9
  }, void 0);
}, as = Ar({ error: "Nco se nepovedlo", success: "Zmna uloena" }), Cu = ({ item: e, attributeName: n, asyncUpdater: r, label: t, type: o = "text" }) => {
  const i = $e(), u = e[n], [s, c] = as(i), f = (m) => {
    const v = { ...e };
    v[n] = m;
    const h = r(v);
    i(h).then(s, c);
  };
  return (
    // <EditableText id={item.id} value={attributeValue} onChange={onChange_} />
    // <TextInput id={item.id} value={attributeValue} onChange={onChange_} />
    /* @__PURE__ */ d.jsxDEV("div", { className: "form-floating", children: [
      /* @__PURE__ */ d.jsxDEV(Ro, { type: o, id: e.id, value: u, onChange: f }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/EditableAttributeText.jsx",
        lineNumber: 36,
        columnNumber: 13
      }, void 0),
      /* @__PURE__ */ d.jsxDEV("label", { htmlFor: e.id, children: t }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/EditableAttributeText.jsx",
        lineNumber: 37,
        columnNumber: 13
      }, void 0)
    ] }, void 0, !0, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/EditableAttributeText.jsx",
      lineNumber: 35,
      columnNumber: 9
    }, void 0)
  );
}, Ru = ({ opName: e, attributeName: n, onChangeFilter: r }) => {
  const [t] = fe(crypto.randomUUID()), [o, i] = fe(""), u = (s) => {
    i(s);
    const c = {};
    c[n] = {}, c[n][e] = s, r && r(s === "" ? null : c);
  };
  return /* @__PURE__ */ d.jsxDEV("div", { className: "form-floating", children: [
    /* @__PURE__ */ d.jsxDEV(Ro, { id: t, value: o, onChange: u }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/TextFilter.jsx",
      lineNumber: 31,
      columnNumber: 13
    }, void 0),
    /* @__PURE__ */ d.jsxDEV("label", { htmlFor: t, children: "Zadejte hodnotu filtru" }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/TextFilter.jsx",
      lineNumber: 33,
      columnNumber: 13
    }, void 0)
  ] }, void 0, !0, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/TextFilter.jsx",
    lineNumber: 30,
    columnNumber: 9
  }, void 0);
}, ss = (e) => (n) => {
  console.log("reduce_or", e, n);
  for (const r of e)
    if (r(n))
      return !0;
  return !1;
}, us = (e) => (n) => {
  for (const r of e)
    if (!r(n))
      return !1;
  return !0;
}, ls = (e, n) => (r) => (console.log("reduce_eq", e, n, r), r[e] === n), cs = (e, n) => (r) => r[e] < n, fs = (e, n) => (r) => r[e] <= n, ds = (e, n) => (r) => r[e] > n, ms = (e, n) => (r) => r[e] >= n, ps = (e, n) => {
  let r = n.replaceAll("%", "");
  return (o) => `${o[e]}`.includes(r);
}, vs = (e, n) => {
  let r = n.replaceAll("%", "");
  return (o) => `${o[e]}`.toLowerCase().includes(r);
}, hs = (e, n) => (r) => n.some((t) => r === t), bs = (e, n) => (r) => `${r[e]}`.startsWith(n), ht = (e, n) => {
  const r = e.map((o) => jo(o));
  return n(r);
}, ys = (e, n) => {
  const r = {
    _eq: ls,
    _like: ps,
    _ilike: vs,
    _lt: cs,
    _le: fs,
    _gt: ds,
    _ge: ms,
    _in: hs,
    _startswith: bs
  }, t = Object.keys(n);
  if (t.length !== 1)
    throw Error(`${n} must have exactly one key`);
  const o = t[0], i = n[o];
  return console.log("createItemFilter", e, o, i), r[o](e, i);
}, jo = (e) => {
  const n = Object.keys(e);
  if (n.length !== 1)
    throw Error(`where filter (${e}) must have exactly one key`);
  let r = () => !1;
  const t = n[0];
  return t === "_or" ? r = ht(e[t], ss) : t === "_and" ? r = ht(e[t], us) : r = ys(t, e[t]), r;
};
var gs = ["color", "size", "title"];
function Nr() {
  return Nr = Object.assign || function(e) {
    for (var n = 1; n < arguments.length; n++) {
      var r = arguments[n];
      for (var t in r)
        Object.prototype.hasOwnProperty.call(r, t) && (e[t] = r[t]);
    }
    return e;
  }, Nr.apply(this, arguments);
}
function Es(e, n) {
  if (e == null)
    return {};
  var r = xs(e, n), t, o;
  if (Object.getOwnPropertySymbols) {
    var i = Object.getOwnPropertySymbols(e);
    for (o = 0; o < i.length; o++)
      t = i[o], !(n.indexOf(t) >= 0) && Object.prototype.propertyIsEnumerable.call(e, t) && (r[t] = e[t]);
  }
  return r;
}
function xs(e, n) {
  if (e == null)
    return {};
  var r = {}, t = Object.keys(e), o, i;
  for (i = 0; i < t.length; i++)
    o = t[i], !(n.indexOf(o) >= 0) && (r[o] = e[o]);
  return r;
}
var Vr = /* @__PURE__ */ Rr(function(e, n) {
  var r = e.color, t = e.size, o = e.title, i = Es(e, gs);
  return /* @__PURE__ */ Ee.createElement("svg", Nr({
    ref: n,
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 16 16",
    width: t,
    height: t,
    fill: r
  }, i), o ? /* @__PURE__ */ Ee.createElement("title", null, o) : null, /* @__PURE__ */ Ee.createElement("path", {
    d: "M1.5 1.5A.5.5 0 0 1 2 1h12a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.128.334L10 8.692V13.5a.5.5 0 0 1-.342.474l-3 1A.5.5 0 0 1 6 14.5V8.692L1.628 3.834A.5.5 0 0 1 1.5 3.5v-2z"
  }));
});
Vr.propTypes = {
  color: F.string,
  size: F.oneOfType([F.string, F.number]),
  title: F.string
};
Vr.defaultProps = {
  color: "currentColor",
  size: "1em",
  title: null
};
const Ns = Vr;
var ws = ["color", "size", "title"];
function wr() {
  return wr = Object.assign || function(e) {
    for (var n = 1; n < arguments.length; n++) {
      var r = arguments[n];
      for (var t in r)
        Object.prototype.hasOwnProperty.call(r, t) && (e[t] = r[t]);
    }
    return e;
  }, wr.apply(this, arguments);
}
function Cs(e, n) {
  if (e == null)
    return {};
  var r = Rs(e, n), t, o;
  if (Object.getOwnPropertySymbols) {
    var i = Object.getOwnPropertySymbols(e);
    for (o = 0; o < i.length; o++)
      t = i[o], !(n.indexOf(t) >= 0) && Object.prototype.propertyIsEnumerable.call(e, t) && (r[t] = e[t]);
  }
  return r;
}
function Rs(e, n) {
  if (e == null)
    return {};
  var r = {}, t = Object.keys(e), o, i;
  for (i = 0; i < t.length; i++)
    o = t[i], !(n.indexOf(o) >= 0) && (r[o] = e[o]);
  return r;
}
var Pr = /* @__PURE__ */ Rr(function(e, n) {
  var r = e.color, t = e.size, o = e.title, i = Cs(e, ws);
  return /* @__PURE__ */ Ee.createElement("svg", wr({
    ref: n,
    xmlns: "http://www.w3.org/2000/svg",
    viewBox: "0 0 16 16",
    width: t,
    height: t,
    fill: r
  }, i), o ? /* @__PURE__ */ Ee.createElement("title", null, o) : null, /* @__PURE__ */ Ee.createElement("path", {
    d: "M1.5 1.5A.5.5 0 0 1 2 1h12a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.128.334L10 8.692V13.5a.5.5 0 0 1-.342.474l-3 1A.5.5 0 0 1 6 14.5V8.692L1.628 3.834A.5.5 0 0 1 1.5 3.5v-2zm1 .5v1.308l4.372 4.858A.5.5 0 0 1 7 8.5v5.306l2-.666V8.5a.5.5 0 0 1 .128-.334L13.5 3.308V2h-11z"
  }));
});
Pr.propTypes = {
  color: F.string,
  size: F.oneOfType([F.string, F.number]),
  title: F.string
};
Pr.defaultProps = {
  color: "currentColor",
  size: "1em",
  title: null
};
const js = Pr, _s = (e, n) => {
  let r = !0, t = n, { _and: o } = t;
  o || (console.error("loadFilters.filter has not _and", n), o = [], t = { _and: [] });
  const i = {};
  for (const s of o) {
    const [c] = Object.keys(s);
    if (c === "_or") {
      const f = s._or.length;
      f < 1 && (console.error("loadFilters.filter._and has too short _or", s._or), r = !1), f > 2 && (console.error("loadFilters.filter._and has too long _or", s._or), r = !1);
      const [m, v] = s._or, [h] = Object.keys(m), [C] = Object.keys(v);
      h !== C && (console.error("loadFilters.filter._and has _or with different names", s._or), r = !1), h in i ? console.error("loadFilters.filter._and has two _or with same names", s._and) : i[h] = s;
    } else
      c in i ? i[c].push(s) : i[c] = [s];
  }
  console.log("reducedFilters", JSON.stringify(i));
  for (const s in i) {
    const c = i[s];
    if (Array.isArray(c)) {
      if (c.length > 2)
        r = !1;
      else
        for (; c.lenght < 2; ) {
          const m = {};
          m[s] = { "": "" }, c.push(m);
        }
      i[s] = { _and: c };
    }
  }
  const { _or: u } = i;
  if (u)
    for (const s of u) {
      const [c, f] = s._or, [m] = Object.keys(c), v = i[m];
      v.lenght !== 0 && (console.error("loadFilters.filter._and has too many filters related to name ", m, v, s), r = !1), v[m] = s;
    }
  return r ? (console.log("loadFilters.valid filter", i), { ...i, _or: null }) : (console.error("loadFilters.invalid filter", i), null);
}, Ss = (e, { attributeName: n, newFilter: r }) => {
  const t = { ...e };
  return t[n] = r, t;
}, Ts = (e) => {
  const n = { _and: [] };
  for (const r in e) {
    if (r === "_or")
      continue;
    const t = e[r], { _or: o, _and: i } = t;
    i && n._and.push(...i), o && n._and.push(t);
  }
  return console.log("getWhereFilter.newOriginalFilter", JSON.stringify(n)), n;
}, Mr = (e, n) => {
  if (n === "")
    return null;
  const r = {};
  if (e) {
    const [t] = Object.values(e) || [""];
    r[n] = t;
  } else
    r[n] = "";
  return r;
}, Vn = (e, n) => {
  const r = {};
  if (e) {
    const [t] = Object.keys(e);
    t ? r[t] = n : r._eq = n;
  } else
    r._eq = n;
  return r;
}, Fs = (e, n) => {
  const r = e || { _and: [{ attributeName: { "": "" } }, { attributeName: { "": "" } }] }, t = {}, o = Object.keys(r);
  let i = null;
  o.length === 1 ? (i = o[0], i !== "_or" & i !== "_and" && (i = "_and"), t[i] = r[i]) : i = "_and";
  const u = r[i];
  if (Array.isArray(u)) {
    const s = [];
    for (const c of u) {
      const f = Object.keys(c);
      if (f.length !== 1)
        throw Error(`Filter must have exactly one key ${JSON.stringify(f)}`);
      const m = f[0];
      if (m !== n)
        continue;
      const v = {};
      v[n] = c[m], s.push(v);
    }
    t[i] = s;
  } else {
    const s = {};
    s[n] = { "": "" };
    const c = {};
    c[n] = { "": "" }, t[i] = [s, c];
  }
  return t;
}, Os = (e, { index: n, attributeName: r, newFilterValue: t }) => {
  const o = {};
  o[r] = { ...t };
  const [i] = Object.keys(e);
  let [u] = Object.values(e);
  n >= u.length ? (console.log("long", n, u.length), u.push(o)) : (console.log("short", n, u.length), u = u.toSpliced(n, 1, o));
  const s = {};
  s[i] = u, console.log("reduced.oldFilter", e), console.log("reduced.newFilter", s);
  const { _or: c, _and: f } = s, v = (c || f).filter((h) => {
    if (h) {
      const [C] = Object.values(h);
      if (!C)
        return console.log("simplefilter", h), !1;
      const [S] = Object.keys(C);
      return !(S === "" || !S);
    }
    return !1;
  }).map((h) => ({ ...h }));
  return console.log("onFilterChange.reduced", v), c ? s._or = v : f && (s._and = v), console.log("onFilterChange.newFilter", s), s;
}, ks = ({ filter: e = { "": "" }, onChange: n }) => {
  const [r] = fe(crypto.randomUUID()), t = (s) => {
    const c = s.target.value, f = Mr(e, c);
    console.log("FilterRow.onChangeOp", c, f), n && n(f);
  }, o = (s) => {
    const c = s.target.value, f = Vn(e, c);
    console.log("FilterRow.onChangeValue", f), n && n(f);
  };
  let i = "", u = "";
  return e && ([i] = Object.keys(e), [u] = Object.values(e)), /* @__PURE__ */ d.jsxDEV(an, { children: [
    /* @__PURE__ */ d.jsxDEV(Ve, { children: /* @__PURE__ */ d.jsxDEV("div", { className: "form-floating", children: [
      /* @__PURE__ */ d.jsxDEV("select", { className: "form-select", id: "select1" + r, value: i, onChange: t, "aria-label": "", children: [
        /* @__PURE__ */ d.jsxDEV("option", { value: "" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
          lineNumber: 41,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_eq", children: "Je rovno" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
          lineNumber: 42,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_gt", children: "Je vt ne" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
          lineNumber: 43,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_ge", children: "Je vt ne nebo rovno" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
          lineNumber: 44,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_lt", children: "Je men ne" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
          lineNumber: 45,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_le", children: "Je men ne nebo rovno" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
          lineNumber: 46,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_ilike", children: "Obsahuje" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
          lineNumber: 47,
          columnNumber: 25
        }, void 0)
      ] }, void 0, !0, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
        lineNumber: 40,
        columnNumber: 21
      }, void 0),
      /* @__PURE__ */ d.jsxDEV("label", { htmlFor: "select1" + r, children: "Opertor" }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
        lineNumber: 49,
        columnNumber: 21
      }, void 0)
    ] }, void 0, !0, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
      lineNumber: 39,
      columnNumber: 17
    }, void 0) }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
      lineNumber: 37,
      columnNumber: 13
    }, void 0),
    /* @__PURE__ */ d.jsxDEV(Ve, { children: /* @__PURE__ */ d.jsxDEV("div", { className: "form-floating", children: [
      /* @__PURE__ */ d.jsxDEV("input", { type: "text", id: "input1" + r, className: "form-control", value: u, onChange: o }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
        lineNumber: 55,
        columnNumber: 21
      }, void 0),
      /* @__PURE__ */ d.jsxDEV("label", { htmlFor: "input1" + r, children: "Hodnota" }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
        lineNumber: 56,
        columnNumber: 21
      }, void 0)
    ] }, void 0, !0, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
      lineNumber: 53,
      columnNumber: 17
    }, void 0) }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
      lineNumber: 52,
      columnNumber: 13
    }, void 0)
  ] }, void 0, !0, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/StringFilterRow.jsx",
    lineNumber: 36,
    columnNumber: 9
  }, void 0);
}, Ds = ({ filter: e = { name: { "": "" } }, onChange: n }) => {
  const [r] = fe(crypto.randomUUID()), t = (s) => {
    const c = s.target.value, f = Mr(e, c);
    console.log("FilterRow.onChangeOp", c, f), n && n(f);
  }, o = (s) => {
    const c = s.target.value, f = Vn(e, +c);
    console.log("FilterRow.onChangeValue", f), n && n(f);
  };
  let i = "", u = "";
  return e && ([i] = Object.keys(e), [u] = Object.values(e)), /* @__PURE__ */ d.jsxDEV(an, { children: [
    /* @__PURE__ */ d.jsxDEV(Ve, { children: /* @__PURE__ */ d.jsxDEV("div", { className: "form-floating", children: [
      /* @__PURE__ */ d.jsxDEV("select", { className: "form-select", id: "select1" + r, value: i, onChange: t, "aria-label": "", children: [
        /* @__PURE__ */ d.jsxDEV("option", { value: "" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
          lineNumber: 40,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_eq", children: "Je rovno" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
          lineNumber: 41,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_gt", children: "Je vt ne" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
          lineNumber: 42,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_ge", children: "Je vt ne nebo rovno" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
          lineNumber: 43,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_lt", children: "Je men ne" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
          lineNumber: 44,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_le", children: "Je men ne nebo rovno" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
          lineNumber: 45,
          columnNumber: 25
        }, void 0)
      ] }, void 0, !0, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
        lineNumber: 39,
        columnNumber: 21
      }, void 0),
      /* @__PURE__ */ d.jsxDEV("label", { htmlFor: "select1" + r, children: "FloatFilterDialog" }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
        lineNumber: 47,
        columnNumber: 21
      }, void 0)
    ] }, void 0, !0, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
      lineNumber: 38,
      columnNumber: 17
    }, void 0) }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
      lineNumber: 37,
      columnNumber: 13
    }, void 0),
    /* @__PURE__ */ d.jsxDEV(Ve, { children: /* @__PURE__ */ d.jsxDEV("div", { className: "form-floating", children: [
      /* @__PURE__ */ d.jsxDEV("input", { type: "text", id: "input1" + r, className: "form-control", value: u, onChange: o }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
        lineNumber: 53,
        columnNumber: 21
      }, void 0),
      /* @__PURE__ */ d.jsxDEV("label", { htmlFor: "input1" + r, children: "Hodnota" }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
        lineNumber: 54,
        columnNumber: 21
      }, void 0)
    ] }, void 0, !0, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
      lineNumber: 51,
      columnNumber: 17
    }, void 0) }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
      lineNumber: 50,
      columnNumber: 13
    }, void 0)
  ] }, void 0, !0, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FloatFilterRow.jsx",
    lineNumber: 36,
    columnNumber: 9
  }, void 0);
}, As = ({ filter: e = { name: { "": "" } }, onChange: n }) => {
  const [r] = fe(crypto.randomUUID()), t = (i) => {
    const u = i.target.value, s = Vn(e, u);
    console.log("FilterRow.onChangeValue", s), n && n(s);
  };
  let o = "";
  return e && ([o] = Object.values(e)), /* @__PURE__ */ d.jsxDEV(an, { children: /* @__PURE__ */ d.jsxDEV(Ve, { children: /* @__PURE__ */ d.jsxDEV("div", { className: "form-floating", children: [
    /* @__PURE__ */ d.jsxDEV("select", { className: "form-select", id: "select1" + r, value: o, onChange: t, "aria-label": "", children: [
      /* @__PURE__ */ d.jsxDEV("option", { value: "" }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/BooleanFilterRow.jsx",
        lineNumber: 28,
        columnNumber: 25
      }, void 0),
      /* @__PURE__ */ d.jsxDEV("option", { value: !0, children: "Pravda" }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/BooleanFilterRow.jsx",
        lineNumber: 29,
        columnNumber: 25
      }, void 0),
      /* @__PURE__ */ d.jsxDEV("option", { value: !1, children: "Nepravda" }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/BooleanFilterRow.jsx",
        lineNumber: 30,
        columnNumber: 25
      }, void 0)
    ] }, void 0, !0, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/BooleanFilterRow.jsx",
      lineNumber: 27,
      columnNumber: 21
    }, void 0),
    /* @__PURE__ */ d.jsxDEV("label", { htmlFor: "select1" + r, children: "FloatFilterDialog" }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/BooleanFilterRow.jsx",
      lineNumber: 32,
      columnNumber: 21
    }, void 0)
  ] }, void 0, !0, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/BooleanFilterRow.jsx",
    lineNumber: 26,
    columnNumber: 17
  }, void 0) }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/BooleanFilterRow.jsx",
    lineNumber: 25,
    columnNumber: 13
  }, void 0) }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/BooleanFilterRow.jsx",
    lineNumber: 24,
    columnNumber: 9
  }, void 0);
}, Vs = ({ filter: e = { name: { "": "" } }, onChange: n }) => {
  const [r] = fe(crypto.randomUUID()), t = (s) => {
    const c = s.target.value, f = Mr(e, c);
    console.log("FilterRow.onChangeOp", c, f), n && n(f);
  }, o = (s) => {
    const c = s.target.value, f = Vn(e, c);
    console.log("FilterRow.onChangeValue", f), n && n(f);
  };
  let i = "", u = "";
  return e && ([i] = Object.keys(e), [u] = Object.values(e)), /* @__PURE__ */ d.jsxDEV(an, { children: [
    /* @__PURE__ */ d.jsxDEV(Ve, { children: /* @__PURE__ */ d.jsxDEV("div", { className: "form-floating", children: [
      /* @__PURE__ */ d.jsxDEV("select", { className: "form-select", id: "select1" + r, value: i, onChange: t, "aria-label": "", children: [
        /* @__PURE__ */ d.jsxDEV("option", { value: "" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
          lineNumber: 39,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_eq", children: "Je rovno" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
          lineNumber: 40,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_gt", children: "Je vt ne" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
          lineNumber: 41,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_ge", children: "Je vt ne nebo rovno" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
          lineNumber: 42,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_lt", children: "Je men ne" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
          lineNumber: 43,
          columnNumber: 25
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("option", { value: "_le", children: "Je men ne nebo rovno" }, void 0, !1, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
          lineNumber: 44,
          columnNumber: 25
        }, void 0)
      ] }, void 0, !0, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
        lineNumber: 38,
        columnNumber: 21
      }, void 0),
      /* @__PURE__ */ d.jsxDEV("label", { htmlFor: "select1" + r, children: "FloatFilterDialog" }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
        lineNumber: 46,
        columnNumber: 21
      }, void 0)
    ] }, void 0, !0, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
      lineNumber: 37,
      columnNumber: 17
    }, void 0) }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
      lineNumber: 36,
      columnNumber: 13
    }, void 0),
    /* @__PURE__ */ d.jsxDEV(Ve, { children: /* @__PURE__ */ d.jsxDEV("div", { className: "form-floating", children: [
      /* @__PURE__ */ d.jsxDEV("input", { type: "date", className: "form-control", id: "input1" + r, value: u, onChange: o }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
        lineNumber: 51,
        columnNumber: 21
      }, void 0),
      /* @__PURE__ */ d.jsxDEV("label", { htmlFor: "input1" + r, children: "Hodnota" }, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
        lineNumber: 53,
        columnNumber: 21
      }, void 0)
    ] }, void 0, !0, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
      lineNumber: 50,
      columnNumber: 17
    }, void 0) }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
      lineNumber: 49,
      columnNumber: 13
    }, void 0)
  ] }, void 0, !0, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/DateFilterRow.jsx",
    lineNumber: 35,
    columnNumber: 9
  }, void 0);
}, Pn = ({ filter: e = { _and: [{ name: { "": "" } }, { name: { "": "" } }] }, attributeName: n = "name", dialogTitle: r = "", onOk: t, onCancel: o, FilterComponent: i }) => {
  const [u, s] = fe(() => Fs(e, n)), c = () => {
    t && t(u);
  }, f = (S) => (T) => {
    const O = Os(u, { index: S, attributeName: n, newFilterValue: T });
    s(O);
  }, m = (S) => {
    const { _or: T, _and: O } = u, A = S.target.id, _ = {};
    T && (_[A] = T), O && (_[A] = O), s(_);
  }, [v] = Object.keys(u);
  let [h, C] = u[v];
  return h && ([h] = Object.values(h)), C && ([C] = Object.values(C)), /* @__PURE__ */ d.jsxDEV(uo, { title: r, onOk: c, onCancel: o, children: [
    JSON.stringify(u),
    /* @__PURE__ */ d.jsxDEV(i, { filter: h, onChange: f(0) }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
      lineNumber: 100,
      columnNumber: 13
    }, void 0),
    /* @__PURE__ */ d.jsxDEV(an, { children: [
      /* @__PURE__ */ d.jsxDEV(Ve, { children: [
        /* @__PURE__ */ d.jsxDEV("div", { className: "form-check form-check-inline", children: [
          /* @__PURE__ */ d.jsxDEV("input", { className: "form-check-input", type: "radio", name: "flexRadioDefault", id: "_and", checked: v === "_and", onChange: m }, void 0, !1, {
            fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
            lineNumber: 104,
            columnNumber: 25
          }, void 0),
          /* @__PURE__ */ d.jsxDEV("label", { className: "form-check-label", htmlFor: "flexRadioDefault1", children: "A" }, void 0, !1, {
            fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
            lineNumber: 105,
            columnNumber: 25
          }, void 0)
        ] }, void 0, !0, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
          lineNumber: 103,
          columnNumber: 21
        }, void 0),
        /* @__PURE__ */ d.jsxDEV("div", { className: "form-check form-check-inline", children: [
          /* @__PURE__ */ d.jsxDEV("input", { className: "form-check-input", type: "radio", name: "flexRadioDefault", id: "_or", checked: v === "_or", onChange: m }, void 0, !1, {
            fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
            lineNumber: 110,
            columnNumber: 25
          }, void 0),
          /* @__PURE__ */ d.jsxDEV("label", { className: "form-check-label", htmlFor: "flexRadioDefault2", children: "Nebo" }, void 0, !1, {
            fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
            lineNumber: 111,
            columnNumber: 25
          }, void 0)
        ] }, void 0, !0, {
          fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
          lineNumber: 109,
          columnNumber: 21
        }, void 0)
      ] }, void 0, !0, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
        lineNumber: 102,
        columnNumber: 17
      }, void 0),
      /* @__PURE__ */ d.jsxDEV(Ve, {}, void 0, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
        lineNumber: 116,
        columnNumber: 17
      }, void 0)
    ] }, void 0, !0, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
      lineNumber: 101,
      columnNumber: 13
    }, void 0),
    /* @__PURE__ */ d.jsxDEV(i, { filter: C, onChange: f(1) }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
      lineNumber: 119,
      columnNumber: 13
    }, void 0)
  ] }, void 0, !0, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
    lineNumber: 97,
    columnNumber: 9
  }, void 0);
}, Ps = ({ filter: e = { _and: [{ name: { "": "" } }, { name: { "": "" } }] }, attributeName: n = "name", dialogTitle: r = "", onOk: t, onCancel: o }) => /* @__PURE__ */ d.jsxDEV(Pn, { filter: e, attributeName: n, dialogTitle: r, onOk: t, onCancel: o, FilterComponent: ks }, void 0, !1, {
  fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
  lineNumber: 128,
  columnNumber: 9
}, void 0), Ms = ({ filter: e = { _and: [{ name: { "": "" } }, { name: { "": "" } }] }, attributeName: n = "name", dialogTitle: r = "", onOk: t, onCancel: o }) => /* @__PURE__ */ d.jsxDEV(Pn, { filter: e, attributeName: n, dialogTitle: r, onOk: t, onCancel: o, FilterComponent: Ds }, void 0, !1, {
  fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
  lineNumber: 134,
  columnNumber: 9
}, void 0), Is = ({ filter: e = { _and: [{ name: { "": "" } }, { name: { "": "" } }] }, attributeName: n = "name", dialogTitle: r = "", onOk: t, onCancel: o }) => /* @__PURE__ */ d.jsxDEV(Pn, { filter: e, attributeName: n, dialogTitle: r, onOk: t, onCancel: o, FilterComponent: As }, void 0, !1, {
  fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
  lineNumber: 140,
  columnNumber: 9
}, void 0), $s = ({ filter: e = { _and: [{ name: { "": "" } }, { name: { "": "" } }] }, attributeName: n = "name", dialogTitle: r = "", onOk: t, onCancel: o }) => /* @__PURE__ */ d.jsxDEV(Pn, { filter: e, attributeName: n, dialogTitle: r, onOk: t, onCancel: o, FilterComponent: Vs }, void 0, !1, {
  fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
  lineNumber: 146,
  columnNumber: 9
}, void 0), Ls = ({ filter: e }) => {
  const n = { Icon: /* @__PURE__ */ d.jsxDEV(js, {}, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
    lineNumber: 152,
    columnNumber: 35
  }, void 0), style: "btn btn-sm btn-outline-primary" }, r = { Icon: /* @__PURE__ */ d.jsxDEV(Ns, {}, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
    lineNumber: 153,
    columnNumber: 37
  }, void 0), style: "btn btn-sm btn-danger" };
  return e ? (e == null ? void 0 : e._and.length) == 0 ? n : r : n;
}, Mn = ({ filter: e, attributeName: n, onChangeFilter: r, Component: t }) => {
  const [o, i] = fe(!1), { Icon: u, style: s } = Ls({ filter: e }), c = (m) => {
    i(!1), r && r(m);
  }, f = () => {
    i(!1);
  };
  return /* @__PURE__ */ d.jsxDEV(d.Fragment, { children: [
    /* @__PURE__ */ d.jsxDEV("span", { className: s, onClick: () => i(!o), children: u }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
      lineNumber: 181,
      columnNumber: 13
    }, void 0),
    o ? /* @__PURE__ */ d.jsxDEV(t, { filter: e, attributeName: n, onOk: c, onCancel: f }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
      lineNumber: 182,
      columnNumber: 22
    }, void 0) : ""
  ] }, void 0, !0, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
    lineNumber: 180,
    columnNumber: 9
  }, void 0);
}, Bs = ({ filter: e, attributeName: n, onChangeFilter: r }) => /* @__PURE__ */ d.jsxDEV(Mn, { filter: e, attributeName: n, onChangeFilter: r, Component: Ps }, void 0, !1, {
  fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
  lineNumber: 189,
  columnNumber: 9
}, void 0), Ws = ({ filter: e, attributeName: n, onChangeFilter: r }) => /* @__PURE__ */ d.jsxDEV(Mn, { filter: e, attributeName: n, onChangeFilter: r, Component: Ms }, void 0, !1, {
  fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
  lineNumber: 195,
  columnNumber: 9
}, void 0), Us = ({ filter: e, attributeName: n, onChangeFilter: r }) => /* @__PURE__ */ d.jsxDEV(Mn, { filter: e, attributeName: n, onChangeFilter: r, Component: Is }, void 0, !1, {
  fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
  lineNumber: 201,
  columnNumber: 9
}, void 0), Ys = ({ filter: e, attributeName: n, onChangeFilter: r }) => /* @__PURE__ */ d.jsxDEV(Mn, { filter: e, attributeName: n, onChangeFilter: r, Component: $s }, void 0, !1, {
  fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
  lineNumber: 207,
  columnNumber: 9
}, void 0), bt = {
  str: Bs,
  num: Ws,
  bool: Us,
  date: Ys
}, qs = ({ columns: e, filter: n, onFilterChange: r }) => {
  const t = (c) => (f) => {
    const m = Ss(o, { attributeName: c, newFilter: f });
    if (i(m), r) {
      const v = Ts(m);
      r(v);
    }
  }, [o, i] = fe(() => _s(null, n)), u = o, s = e.map((c) => ({ type: c.type, label: c.label, property: c.property }));
  return /* @__PURE__ */ d.jsxDEV("tr", { children: s.map(
    (c, f) => {
      let m = null;
      return c.type in bt && (m = bt[c.type]), /* @__PURE__ */ d.jsxDEV("th", { children: [
        c.label,
        " ",
        m ? /* @__PURE__ */ d.jsxDEV(
          m,
          {
            filter: c.property in u ? u[c.property] : null,
            attributeName: c.property,
            onChangeFilter: t(c.property)
          },
          void 0,
          !1,
          {
            fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
            lineNumber: 248,
            columnNumber: 52
          },
          void 0
        ) : ""
      ] }, f, !0, {
        fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
        lineNumber: 247,
        columnNumber: 25
      }, void 0);
    }
  ) }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Filters/FilterManager.jsx",
    lineNumber: 238,
    columnNumber: 9
  }, void 0);
}, Hs = "/api/gql", yt = {
  method: "POST",
  headers: {
    "Content-Type": "application/json"
  },
  cache: "no-cache",
  // *default, no-cache, reload, force-cache, only-if-cached
  redirect: "follow"
  // manual, *follow, error
}, zs = (e, n) => {
  console.log("fetch from shared");
  const r = { ...yt.headers, ...n.headers }, t = { ...yt, ...n, headers: r };
  return t.body = t.body.replaceAll(": ID", ": UUID"), fetch(Hs, t).then((o) => o.json());
}, Js = (e, n) => ({ query: e, variables: n }), Ir = (e, n = {}) => (r) => {
  const t = JSON.stringify(Js(e, r));
  return zs("", { body: t, ...n });
}, ju = Ir, _o = (e, n = {}) => {
  if (typeof e != "string")
    throw new Error("CreateAsyncActionFromQuery query param have be string!");
  const r = Ir(e, n);
  return (t) => async (o) => {
    const i = await r(t), u = i == null ? void 0 : i.data;
    if (u) {
      const s = u == null ? void 0 : u.result;
      s && (Array.isArray(s) ? s.forEach((c) => {
        o(Er.item_update(c));
      }) : o(Er.item_update(s)));
    }
    return i;
  };
}, _u = (e, n = {}) => {
  if (typeof e != "string")
    throw new Error("CreateAsyncActionFromMutation query param have be string!");
  const r = Ir(e, n);
  return (t) => async (o) => {
    const i = await r(t), u = i == null ? void 0 : i.data;
    if (u) {
      const s = u == null ? void 0 : u.result;
      if (s) {
        const c = s == null ? void 0 : s.result;
        c && o(Er.item_update(c));
      }
    }
    return i;
  };
}, Gs = `{
    result: me {
      __typename
      id
      name
      surname
      fullname
      email
      roles {
        group {
          id
          name
        }
        roletype {
          id
          name
        }
      }
    }
  }`, Ks = _o(Gs), Xs = ({ loginURL: e = "/oauth/login2?redirect_uri=/", logoutURL: n = "/oauth/logout" }) => {
  const r = $e(), [t, o] = fe(null);
  return Se(() => {
    (async () => {
      var c;
      const u = await r(Ks({})), s = (c = u == null ? void 0 : u.data) == null ? void 0 : c.result;
      s && o(s);
    })();
  }, [r, o]), t ? /* @__PURE__ */ d.jsxDEV("div", { className: "navbar-nav text-end m-0 p-0", children: /* @__PURE__ */ d.jsxDEV("a", { className: "nav-link", href: n, children: t == null ? void 0 : t.fullname }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Me.jsx",
    lineNumber: 47,
    columnNumber: 17
  }, void 0) }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Me.jsx",
    lineNumber: 46,
    columnNumber: 13
  }, void 0) : /* @__PURE__ */ d.jsxDEV("div", { className: "navbar-nav text-end m-0 p-0", children: /* @__PURE__ */ d.jsxDEV("a", { className: "nav-link", href: e, children: "Login" }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Me.jsx",
    lineNumber: 53,
    columnNumber: 17
  }, void 0) }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Components/Me.jsx",
    lineNumber: 52,
    columnNumber: 13
  }, void 0);
};
new Promise(() => 0);
const Zs = ({ id: e }, n) => {
  const r = $e(), t = An((s) => s.items);
  if (!t)
    throw Error("bad use of store and useFreshItem hook, checks that store state has items attribute");
  const o = t[e], [i, u] = fe(new Promise(() => 0));
  return Se(
    () => {
      const c = new AbortController().signal;
      u((f) => r(n({ id: e }, c)));
    },
    [e, n, r]
  ), [o, i];
}, Su = ({ id: e }, n) => {
  const r = $e();
  return [An((i) => i.items)[e], r(n({ id: e }))];
};
/**
 * @remix-run/router v1.14.1
 *
 * Copyright (c) Remix Software Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.md file in the root directory of this source tree.
 *
 * @license MIT
 */
var gt;
(function(e) {
  e.Pop = "POP", e.Push = "PUSH", e.Replace = "REPLACE";
})(gt || (gt = {}));
function Et(e, n) {
  if (e === !1 || e === null || typeof e > "u")
    throw new Error(n);
}
var xt;
(function(e) {
  e.data = "data", e.deferred = "deferred", e.redirect = "redirect", e.error = "error";
})(xt || (xt = {}));
const So = ["post", "put", "patch", "delete"];
new Set(So);
const Qs = ["get", ...So];
new Set(Qs);
/**
 * React Router v6.21.1
 *
 * Copyright (c) Remix Software Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.md file in the root directory of this source tree.
 *
 * @license MIT
 */
const eu = /* @__PURE__ */ U.createContext(null);
process.env.NODE_ENV !== "production" && (eu.displayName = "DataRouter");
const nu = /* @__PURE__ */ U.createContext(null);
process.env.NODE_ENV !== "production" && (nu.displayName = "DataRouterState");
const ru = /* @__PURE__ */ U.createContext(null);
process.env.NODE_ENV !== "production" && (ru.displayName = "Await");
const tu = /* @__PURE__ */ U.createContext(null);
process.env.NODE_ENV !== "production" && (tu.displayName = "Navigation");
const $r = /* @__PURE__ */ U.createContext(null);
process.env.NODE_ENV !== "production" && ($r.displayName = "Location");
const To = /* @__PURE__ */ U.createContext({
  outlet: null,
  matches: [],
  isDataRoute: !1
});
process.env.NODE_ENV !== "production" && (To.displayName = "Route");
const ou = /* @__PURE__ */ U.createContext(null);
process.env.NODE_ENV !== "production" && (ou.displayName = "RouteError");
function iu() {
  return U.useContext($r) != null;
}
function au() {
  return iu() || (process.env.NODE_ENV !== "production" ? Et(
    !1,
    // TODO: This error is probably because they somehow have 2 versions of the
    // router loaded. We can help them understand how to avoid that.
    "useLocation() may be used only in the context of a <Router> component."
  ) : Et(!1)), U.useContext($r).location;
}
function su() {
  let {
    matches: e
  } = U.useContext(To), n = e[e.length - 1];
  return n ? n.params : {};
}
new Promise(() => {
});
const Tu = () => {
  const { pathname: e } = au(), { id: n } = su();
  let r = e, t = !1;
  e.includes("edit") ? (r = e.replaceAll("edit/" + n, n), t = !0) : r = e.replaceAll(n, "edit/" + n);
  const o = {
    editlink: r,
    editting: t,
    linkto: ({ name: i, id: u }) => {
      const s = e.split("/");
      return s[1] = i, s[s.length - 1] = `${u}`, s.join("/");
    }
  };
  return console.log("usePath", e, n), console.log("usePath:", o.linkto({ name: "users", id: "1202fc8d-fe0e-41a6-96a6-2009738bc62a" })), o;
}, uu = ({ children: e }) => {
  const n = Ao({
    reducer: {
      items: za,
      msgs: rs
    },
    preloadedState: {
      items: {},
      msgs: {}
    }
  });
  return /* @__PURE__ */ d.jsxDEV(Aa, { store: n, children: e }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/Store/AppProvider.jsx",
    lineNumber: 26,
    columnNumber: 9
  }, void 0);
}, lu = `query ($skip: Int, $limit: Int, $where: UserInputWhereFilter, $orderby: String) {
    result: userPage(skip: $skip, limit: $limit, where: $where, orderby: $orderby) {
      __typename
      id
      name
      surname
      fullname
      email
    }
  }`, cu = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJ0b2tlbl90eXBlIjoiQmVhcmVyIiwiYWNjZXNzX3Rva2VuIjoiQUNDVC02MUVseG5oVHlhRVN4RkFna2lvb3AwdjF5WG96T0xUSiIsImV4cGlyZXNfaW4iOjM2MDAsInJlZnJlc2hfdG9rZW4iOiJSRUZULU5SQmRDekdwb09SMENIMWo3ZVBGdFRRWFVQcEtmdmxoIiwidXNlcl9pZCI6IjJkOWRjNWNhLWE0YTItMTFlZC1iOWRmLTAyNDJhYzEyMDAwMyJ9.K0UJ4RaHUrgl-HewYTcEBRHZ677ieGXEPbnfNn6gbOX-5o6A_u19mtr7IjuiX_DNXdDG4KdiC3mZ7GkOlMgQvO2N0lHlSFd2f8I-zI_UgQYfEYR33vGuSQC1KGv_hrTO2Cn7wUZen33Mz6DvcPrUPAaw86izf4pliKUX28lhRnIaRKIUWmVUAKumJxBVJZzDtEaG68rXT6PGkJ4zzN0eqCOgIfq2vDXRpNBlvFQ8rok2MRLBnV2H_IlKskpebiRck8qDVdJ81__IvBdNo7e3dZyvlSYBemPn96j_58WE--99_4v6BqfdBdkLDea7yLxz2IMCeygVDo2h4wyYwWmBKQ", fu = { authorization: `Bearer ${cu}` }, Cr = _o(lu, { headers: fu }), Fo = Ar({ error: "Nepovedlo se nast uivatele", success: "Naten uivatel se povedlo" }), du = ({ onChangeFilter: e }) => {
  fe({});
  const n = (r) => {
    console.log("UsersTableHeader.onFilterChange.filter", r), e(r);
  };
  return /* @__PURE__ */ d.jsxDEV("thead", { children: /* @__PURE__ */ d.jsxDEV(
    qs,
    {
      columns: [
        { label: "#", property: null, type: null },
        { label: "Jmno", property: "name", type: "str" },
        { label: "Pjmen", property: "surname", type: "str" },
        { label: "Email", property: "email", type: "str" }
      ],
      filter: { _and: [{ name: { _eq: "John" } }, { surname: { _gt: "A" } }, { surname: { _lt: "Z" } }] },
      onFilterChange: n
    },
    void 0,
    !1,
    {
      fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
      lineNumber: 87,
      columnNumber: 13
    },
    void 0
  ) }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 74,
    columnNumber: 9
  }, void 0);
}, mu = ({ user: e, index: n }) => /* @__PURE__ */ d.jsxDEV("tr", { children: [
  /* @__PURE__ */ d.jsxDEV("td", { children: n }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 104,
    columnNumber: 13
  }, void 0),
  /* @__PURE__ */ d.jsxDEV("td", { children: e.name }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 105,
    columnNumber: 13
  }, void 0),
  /* @__PURE__ */ d.jsxDEV("td", { children: e.surname }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 106,
    columnNumber: 13
  }, void 0),
  /* @__PURE__ */ d.jsxDEV("td", { children: e.email }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 107,
    columnNumber: 13
  }, void 0)
] }, void 0, !0, {
  fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
  lineNumber: 103,
  columnNumber: 9
}, void 0), pu = ({ users: e }) => {
  const [n, r] = fe(0), [t, o] = fe(null), [i, u] = fe(!0), s = 10, c = $e(), [f, m] = Fo($e()), v = (O) => {
    console.log("UsersTable.onChangeFilter", O), o(() => O), h(O), u(!0), r(0);
  }, h = (O = t) => {
    r(n + s);
    let A = null;
    O ? A = c(Cr({ skip: n + s, where: O })) : A = c(Cr({ skip: n + s })), A.then(f, m).then((_) => {
      var q;
      (((q = _ == null ? void 0 : _.data) == null ? void 0 : q.result) || []).length == 0 && u(!1);
    }, (_) => {
      console.log("error", _);
    });
  }, C = () => {
    h(t);
  }, S = i ? /* @__PURE__ */ d.jsxDEV("tfoot", { children: /* @__PURE__ */ d.jsxDEV("tr", { children: /* @__PURE__ */ d.jsxDEV("th", { colSpan: 5, children: /* @__PURE__ */ d.jsxDEV("button", { className: "btn btn-outline-primary", onClick: C, children: "Nast dal" }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 205,
    columnNumber: 25
  }, void 0) }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 204,
    columnNumber: 21
  }, void 0) }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 203,
    columnNumber: 17
  }, void 0) }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 202,
    columnNumber: 9
  }, void 0) : "";
  let T = e;
  return t && (T = e.filter(jo(t))), /* @__PURE__ */ d.jsxDEV("table", { className: "table table-striped table-bordered", children: [
    /* @__PURE__ */ d.jsxDEV(du, { onChangeFilter: v }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
      lineNumber: 221,
      columnNumber: 13
    }, void 0),
    /* @__PURE__ */ d.jsxDEV("tbody", { children: T.map(
      (O, A) => /* @__PURE__ */ d.jsxDEV(mu, { user: O, index: A }, O.id, !1, {
        fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
        lineNumber: 224,
        columnNumber: 31
      }, void 0)
    ) }, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
      lineNumber: 222,
      columnNumber: 13
    }, void 0),
    S
  ] }, void 0, !0, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 220,
    columnNumber: 9
  }, void 0);
}, vu = () => {
  const [e, n] = Zs({ id: "idontcare" }, Cr), [r, t] = Fo($e());
  console.log("AppBody.render", n), n.then(r, t);
  const o = An((u) => u.items), i = Object.values(o).filter((u) => (u == null ? void 0 : u.__typename) === "UserGQLModel");
  return i ? /* @__PURE__ */ d.jsxDEV("div", { children: /* @__PURE__ */ d.jsxDEV(pu, { users: i }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 256,
    columnNumber: 17
  }, void 0) }, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 255,
    columnNumber: 13
  }, void 0) : /* @__PURE__ */ d.jsxDEV("div", { className: "card", children: [
    JSON.stringify(o),
    /* @__PURE__ */ d.jsxDEV("br", {}, void 0, !1, {
      fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
      lineNumber: 261,
      columnNumber: 56
    }, void 0),
    "Loading ..."
  ] }, void 0, !0, {
    fileName: "/home/runner/work/frontendui/frontendui/src/AppBody.jsx",
    lineNumber: 261,
    columnNumber: 11
  }, void 0);
}, hu = ({ children: e }) => /* @__PURE__ */ d.jsxDEV("div", { className: "container-fluid", children: /* @__PURE__ */ d.jsxDEV(uu, { children: [
  /* @__PURE__ */ d.jsxDEV(os, {}, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/App.jsx",
    lineNumber: 13,
    columnNumber: 17
  }, void 0),
  e
] }, void 0, !0, {
  fileName: "/home/runner/work/frontendui/frontendui/src/App.jsx",
  lineNumber: 12,
  columnNumber: 13
}, void 0) }, void 0, !1, {
  fileName: "/home/runner/work/frontendui/frontendui/src/App.jsx",
  lineNumber: 11,
  columnNumber: 9
}, void 0), Fu = () => /* @__PURE__ */ d.jsxDEV(hu, { children: [
  /* @__PURE__ */ d.jsxDEV(Xs, {}, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/App.jsx",
    lineNumber: 23,
    columnNumber: 13
  }, void 0),
  /* @__PURE__ */ d.jsxDEV(vu, {}, void 0, !1, {
    fileName: "/home/runner/work/frontendui/frontendui/src/App.jsx",
    lineNumber: 24,
    columnNumber: 13
  }, void 0)
] }, void 0, !0, {
  fileName: "/home/runner/work/frontendui/frontendui/src/App.jsx",
  lineNumber: 22,
  columnNumber: 9
}, void 0);
export {
  Fu as App,
  hu as AppCanvas,
  Us as BooleanFilterButton,
  Is as BooleanFilterDialog,
  Eu as CardCapsule,
  Nu as CheckGQLError,
  _u as CreateAsyncActionFromMutation,
  _o as CreateAsyncActionFromQuery,
  Ar as CreateAsyncQueryValidator,
  ei as CreateDelayer,
  ju as CreateFetchQuery,
  mo as CreateItem,
  Ys as DateFilterButton,
  $s as DateFilterDialog,
  xu as DeleteButton,
  po as DeleteItem,
  uo as Dialog,
  wu as EditableAttributeSelect,
  Cu as EditableAttributeText,
  Ws as FloatFilterButton,
  Ms as FloatFilterDialog,
  Er as ItemActions,
  za as ItemReducer,
  vo as ItemSlice,
  Xs as LogButton,
  Fn as MsgActions,
  fr as MsgAddAction,
  vt as MsgFlashAction,
  rs as MsgReducer,
  os as Msgs,
  Ma as ReplaceItem,
  Bs as StringFilterButton,
  Ps as StringFilterDialog,
  qs as TableHeaderWithFilters,
  Ru as TextFilter,
  Ro as TextInput,
  Ia as UpdateItem,
  La as UpdateSubScalar,
  $a as UpdateSubVector,
  zs as authorizedFetch2,
  $e as useDispatch,
  Zs as useFreshItem,
  Su as useFreshItem_,
  Tu as usePath,
  jo as where2filter
};
